<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Collecting primary data</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IaSC meta-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="primary_data.html">Primary studies</a>
</li>
<li>
  <a href="meta_analysis.html">Meta-analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/exp_evol_metaanalysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Collecting primary data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>exp_evol_metaanalysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20260206code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20260206)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20260206code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20260206)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanexpevolmetaanalysistree41e5d79504463eea6914af5f2360344caf3b672ftargetblank41e5d79a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/tree/41e5d79504463eea6914af5f2360344caf3b672f" target="_blank">41e5d79</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanexpevolmetaanalysistree41e5d79504463eea6914af5f2360344caf3b672ftargetblank41e5d79a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/tree/41e5d79504463eea6914af5f2360344caf3b672f" target="_blank">41e5d79</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    input_data/.DS_Store
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/.DS_Store
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/DevelopmentTimeGeneration/.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/LocomotionAssayGeneration123/.DS_Store
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/LocomotionAssayGeneration123/.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/LocomotionAssayGeneration123/.Rhistory
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/ReproductiveFitnessFM/.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/ReproductiveFitnessFM/.Rhistory
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/StandardFitnessGeneration15-18/.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/StandardFitnessGeneration39-41 /.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/ThoraxSizeGeneration15-18/.Rapp.history
    Ignored:    input_data/lundhansen_github_data_FLXevoexp-master/ThoraxSizeGeneration72/.Rapp.history

Untracked files:
    Untracked:  effect_sizes/jiang_2011_1.csv
    Untracked:  effect_sizes/jiang_2011_2.csv
    Untracked:  effect_sizes/jiang_2011_3.csv
    Untracked:  effect_sizes/jiang_2011_4.csv
    Untracked:  effect_sizes/prasad2007_1.csv
    Untracked:  effect_sizes/prasad2007_2.csv
    Untracked:  effect_sizes/prasad2007_3.csv
    Untracked:  effect_sizes/prasad2007_4.csv
    Untracked:  effect_sizes/prasad2007_5.csv
    Untracked:  effect_sizes/prasad2007_6.csv

Unstaged changes:
    Deleted:    data/README.md
    Modified:   effect_sizes/abbott2010_1.csv
    Modified:   effect_sizes/abbott2010_2.csv
    Modified:   effect_sizes/abbott2010_5.csv
    Modified:   effect_sizes/abbott2010_6.csv
    Modified:   effect_sizes/abbott_2013_2.csv
    Deleted:    effect_sizes/martinossiAllibert2019_1.csv
    Deleted:    effect_sizes/martinossiAllibert2019_2.csv
    Modified:   effect_sizes/melo_gavin2025_maleharm.csv
    Modified:   effect_sizes/norden2020.csv
    Modified:   input_data/study_metadata.csv
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/primary_data.Rmd</code>) and HTML
(<code>docs/primary_data.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/41e5d79504463eea6914af5f2360344caf3b672f/analysis/primary_data.Rmd" target="_blank">41e5d79</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/c218b224d367cd0ea959f9b9f8566d4667ea2bc9/docs/primary_data.html" target="_blank">c218b22</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/bf1e9d33f61a9f1c656950962ec8631cd6f5ca7f/analysis/primary_data.Rmd" target="_blank">bf1e9d3</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/674ad007f9126d93f696e8995f7edc17792b6b34/docs/primary_data.html" target="_blank">674ad00</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/3e2b8938bd98e8390385d5564885f78776e5f849/analysis/primary_data.Rmd" target="_blank">3e2b893</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/1c0d55ca2fd9fec647bc98423d67f8dba378bb27/docs/primary_data.html" target="_blank">1c0d55c</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/a3a96cb0e1a27732a2dc655b657fe3835c465954/analysis/primary_data.Rmd" target="_blank">a3a96cb</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/94ac9a6cbff9c648f060e24c454371ca769790d2/docs/primary_data.html" target="_blank">94ac9a6</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/5ad4bc9b00ca18d74c637d75a8b6318f8feccaad/analysis/primary_data.Rmd" target="_blank">5ad4bc9</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/145ea3c19e18b31dc92ff7ed78616ca5bf28243d/docs/primary_data.html" target="_blank">145ea3c</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/d0a61c00508a47dbd1555d5dc432d12d36f8abed/analysis/primary_data.Rmd" target="_blank">d0a61c0</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/21970e60f54989fc965c887bff97f04a5e56cd43/docs/primary_data.html" target="_blank">21970e6</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/1879f8338da34e3f16501a55345a308fcbbd638f/analysis/primary_data.Rmd" target="_blank">1879f83</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-26
</td>
<td>
wflow_publish("analysis/primary_data.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/39db66a28cdaaac6710509fe07865f3cc0aed778/docs/primary_data.html" target="_blank">39db66a</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/9cb6a11cf325f1b36b5fa2e7318b9a0e50b765b1/analysis/primary_data.Rmd" target="_blank">9cb6a11</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-25
</td>
<td>
Edit trait name
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/a6905dafa3f16d82b48652c43151b24348ef8216/docs/primary_data.html" target="_blank">a6905da</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/a6bf3dadd32bbf1c28f1d14194af95e22c91816e/analysis/primary_data.Rmd" target="_blank">a6bf3da</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-25
</td>
<td>
Edit Melo-Gavin 2025 and delete M-R
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/4719bf8fc197e2d0fa097114f3efa09cea71e432/docs/primary_data.html" target="_blank">4719bf8</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/b781d8214c2d5b9cc0a67996615495c8fa15227c/analysis/primary_data.Rmd" target="_blank">b781d82</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-20
</td>
<td>
Add Melo-Gavin 2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/1b001bcc54918afb7d40df3aecf5ce20fb51a4f4/docs/primary_data.html" target="_blank">1b001bc</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/61f48377e7a1e99c22aa7c43a1c5512058bdb797/analysis/primary_data.Rmd" target="_blank">61f4837</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-20
</td>
<td>
Fix typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/34e7ec6a1a2e98e0461e9c0d6a89e0b1092f2cc1/docs/primary_data.html" target="_blank">34e7ec6</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/eb8d223dc7f4c57849d147e1b64b8aec06efaf94/analysis/primary_data.Rmd" target="_blank">eb8d223</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-10
</td>
<td>
Fix typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/f546a4caff8604bfc3960c9234d49f1cb6823a43/docs/primary_data.html" target="_blank">f546a4c</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/2e563e5af3ff960ff360740c4ff716196bdbaf6c/analysis/primary_data.Rmd" target="_blank">2e563e5</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-10
</td>
<td>
Fixed a few bits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/ed814c11aafb2319e6e2f9a173420bd93df78635/docs/primary_data.html" target="_blank">ed814c1</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/7f4053a56e1d4ce7a7fde06d359f44a97c8e7859/analysis/primary_data.Rmd" target="_blank">7f4053a</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
fix typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/0b01f666116032434912cf3eaf8a85cdcfd7710d/docs/primary_data.html" target="_blank">0b01f66</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/27f045946747f8ffcc8adb03b6587c574c6e0f44/docs/primary_data.html" target="_blank">27f0459</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/7d7483cda5459177460914e408443e24c931c24c/docs/primary_data.html" target="_blank">7d7483c</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/b0251e46003e3d313cc1d9ad34d9974bb136ba26/analysis/primary_data.Rmd" target="_blank">b0251e4</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Many minor edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/b0251e46003e3d313cc1d9ad34d9974bb136ba26/docs/primary_data.html" target="_blank">b0251e4</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Many minor edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/f4fb206991bca3ddc6ae88c2dc65680194228b88/docs/primary_data.html" target="_blank">f4fb206</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
First commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/f1a4d10896b3997ea3a85d492f6ada1dda89bd50/docs/primary_data.html" target="_blank">f1a4d10</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
First commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/6029ddfff62d25282a0b2966449edc5d789c2c0c/docs/primary_data.html" target="_blank">6029ddf</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/exp_evol_metaanalysis/blob/7057ba1a28bb7ccd9001d258a29f881a16c9bc02/analysis/primary_data.Rmd" target="_blank">7057ba1</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
First commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/exp_evol_metaanalysis/7057ba1a28bb7ccd9001d258a29f881a16c9bc02/docs/primary_data.html" target="_blank">7057ba1</a>
</td>
<td>
luekholman
</td>
<td>
2026-02-09
</td>
<td>
First commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(readxl)
library(lme4)
library(car)
library(lmerTest)
library(emmeans)
library(survival)
library(kableExtra)
library(compute.es)

# Function to parse results of compute.es::mes(), fes(), etc.
parse_es &lt;- function(x, study, trait, generations, organism, interpretation){
  x %&gt;% 
    as.list() %&gt;% enframe() %&gt;% unnest(value) %&gt;% 
    filter(name %in% c(&quot;d&quot;, &quot;var.d&quot;, &quot;l.d&quot;, &quot;u.d&quot;)) %&gt;% 
    mutate(type = c(&quot;Estimate&quot;, &quot;Var&quot;, &quot;Lower_95_CI&quot;, &quot;Upper_95_CI&quot;)) %&gt;% 
    select(-name) %&gt;% 
    pivot_wider(names_from = type, values_from = value) %&gt;% 
    mutate(Study = study, Trait = trait, Generations = generations, 
           Organism = organism, Interpretation = interpretation, 
           .before = 1)
}

# Function to calculate Cohen&#39;s d effect size from a 2x2 contingency table
# It first calculates the log odds ratio, then converts this to Cohen&#39;s d via logOR * sqrt(3) / pi
# SE of the LOR is calculated using Woolf&#39;s method
get_effect_size_2x2_matrix &lt;- function(x){
  
  # calculate log odds ratio as log(a*d / b*c)
  logOR &lt;- as.numeric(log((x[1,1] * x[2,2]) / 
                            (x[2,1] * x[1,2])))
  
  # Standard error of the log OR (Woolf method)
  SE_logOR &lt;- 1/x[1,1] + 1/x[1,2] + 1/x[2,1] + 1/x[2,2]
  
  # 95% CIs of the log OR
  CI_logOR &lt;- logOR + c(-1, 1) * 1.96 *  SE_logOR
  
  # Convert LOR to d:
  d &lt;- logOR * sqrt(3) / pi
  SE_d &lt;- SE_logOR * sqrt(3) / pi
  CI_d &lt;- CI_logOR * sqrt(3) / pi
  
  tibble(Estimate = d,
         Var = SE_d ^ 2,
         Lower_95_CI = CI_d[1],
         Upper_95_CI = CI_d[2])
}</code></pre>
<div id="overview" class="section level1">
<h1>Overview</h1>
<div id="cohens-d" class="section level2">
<h2>Cohen’s <span class="math inline">\(d\)</span></h2>
<p>This document describes how I estimated Cohen’s <span
class="math inline">\(d\)</span> for each primary study in the
meta-analysis. Cohen’s <span class="math inline">\(d\)</span> is a
standardised measure of the effect size, defined as the difference
between two means, expressed in standard deviations of the variable of
interest. Conventionally, Cohen’s <span class="math inline">\(d &gt;
|0.5|\)</span> is considered a ‘large’ effect.</p>
</div>
<div id="meaning-of-effect-size-sign" class="section level2">
<h2>Meaning of effect size sign</h2>
<p>Crucially, effect sizes have a sign (positive or negative, depending
on whether mean 1 or mean 2 is the larger) as well as a magnitude.
Unlike the magnitude, the sign is determined solely by the researcher:
it depends on which of the two means is designated as the reference for
comparison.</p>
<p>To ensure the effect size sign is meaningful and comparable across
studies, <strong>I defined a positive effect size to indicate that the
two means differ in the direction that one would predict if intralocus
sexual conflict is present</strong>, and a negative effect to mean that
the means differ in a direction opposite to this prediction.</p>
<p>For example, for a study like Prasad et al. (2007) that removed
selection on females and then measured mean male and female fitness
relative to a control, we would predict male-evolved populations to have
<em>higher</em> male fitness and <em>lower</em> female fitness if
intralocus sexual conflict is present. So, a positive effect size should
indicate higher male fitness but lower female fitness in the
experimental group relative to the control. This is potentially
confusing, but it ensures that finding the mean effect size via
meta-analysis is useful: a positive mean effect size would mean that
experimental evolution studies tend to produce the results predicted if
IaSC is common.</p>
<p>To clarify the meaning of each effect size, I provide a column called
‘interpretation’ in the final table of effect sizes, and document the
more confusing/ambiguous decisions in this document below.</p>
</div>
<div id="how-cohens-d-was-calculated" class="section level2">
<h2>How Cohen’s <span class="math inline">\(d\)</span> was
calculated</h2>
<div id="preferred-method-from-the-raw-data" class="section level3">
<h3>Preferred method: from the raw data</h3>
<p>If the primary study’s raw data were available (e.g. on Dryad or
Figshare, or as a table in the original paper), I calculated the effect
size from an appropriate statistical model that accounted for the
structure of the experimental design (e.g. the use of replicate
experimental evolution lines), block effects, etc. (following the model
used by the original authors wherever possible). The models chosen are
documented in the code below.</p>
<p>For response variables where it was appropriate to use Gaussian
linear (mixed) models, I first transformed the response variable to have
a mean of zero and variance of 1, such that the difference between
treatment means estimates by the statistical model is equal to Cohen’s
d. </p>
<p>For generalised (e.g. binomial) models that estimated treatment means
as a log odds ratio (<span
class="math inline">\(\log(\mathrm{OR})\)</span>), or a log hazard ratio
(for survival models), I converted this effect size to Cohen’s d using
the standard approximation:</p>
<p><span class="math inline">\(\log(\mathrm{OR}) =
\frac{\sqrt{3}}{\pi}\)</span></p>
<p>For studies where the original data represent counts of binary events
tallied up in a <span class="math inline">\(2\times2\)</span>
contingency table (e.g. number of surviving and non-surviving
individuals from the treatment and control groups) with elements <span
class="math inline">\(a\)</span>,<span
class="math inline">\(b\)</span>,<span class="math inline">\(c\)</span>
and <span class="math inline">\(d\)</span>, I estimated the log odds
ratio as <span class="math inline">\(log(a\times d / b\times c)\)</span>
and found the SE of the log odds ratio using Woolf’s method, i.e. </p>
<p><span class="math display">\[SE\!\left(\log(\mathrm{OR})\right)
=\sqrt{\frac{1}{a}+\frac{1}{b}+\frac{1}{c}+\frac{1}{d}}\]</span></p>
</div>
<div id="fall-back-method-from-the-group-means" class="section level3">
<h3>Fall-back method: from the group means</h3>
<p>For many studies, the raw data were not available, but the paper
provided numerical or graphical information about 1) the means of each
treatment group, 2) either the group standard deviations or the standard
errors of the means (which allow calculation of the standard deviation
as <span class="math inline">\(SD = SE\times\sqrt{N}\)</span>), and 3)
the sample size N. From these three quantities, one can estimate Cohen’s
d (I used the function <code>mes()</code> from the
<code>compute.es</code> package). Where possible I obtained these
numbers from tables or the main text, but often the data were only
contained in figures (e.g. bar charts showing the mean and standard
errors). In the latter case, I measured the means and standard errors
using the web application <a
href="https://plotdigitizer.com/">PlotDigitizer</a> - this data
extraction step is expected to add inaccuracy, but not to add consistent
bias.</p>
<p>Using group means is a less accurate method because it is not
possible to account for the structure of the experimental design,
incurring pseudoreplication. For example, in studies where the authors
had 4 replicate experimental evolution lines for 2 treatments and
measured 20 individuals per line, the sample size is not really 80 per
treatment, since there are 20 replicate measurements of only 8 lines.
This will make the effect size measurements appear more precise than
they should be, giving extra weight to those studies in the
meta-analysis.</p>
</div>
<div id="workflow-in-the-code-below" class="section level3">
<h3>Workflow in the code below</h3>
<p>For each publication, I either load or type out the raw data, display
it (in cases where I extracted the raw data or group means from the
primary paper), calculate effect size, and then save it to a small .csv
file along with meta-data like the paper of origin, the trait being
measured, the number of generations of selection prior to measuring the
trait, and a plain English interpretation of the meaning of the effect
size (e.g. that the trait is higher/lower in one treatment group
compared to another such group).</p>
</div>
</div>
</div>
<div id="sex-limited-evolution-experiments" class="section level1">
<h1>Sex-limited evolution experiments</h1>
<p>This section covers publications that evolved genomic elements
(e.g. small regions or whole chromosomes) mostly/entirely within one sex
for multiple generations. In general, I predict that this treatment will
increase fitness in that sex, reduce fitness in the other sex, and cause
phenotypic evolution towards the typical trait values for the evolving
sex (e.g. in <em>Drosophila</em>, chromosomes that had undergone
male-limited evolution in should encode smaller body size and slower
development time). The effect size signs selected reflect these
expectations.</p>
<div id="rice-1992" class="section level2">
<h2>Rice (1992)</h2>
<p>Rice, W. R. (1992). Sexually antagonistic genes: experimental
evidence. <em>Science</em>, 256, 1436-1439.</p>
<div id="rices-original-data" class="section level3">
<h3>Rice’s original data</h3>
<p>These data are shown in Table 1. Letters R, P, S and O denote the 4
eye colours of males that females might mate to, where O (orange-eyed)
represents non-focal males and R/P/S are the focal male (which differ in
how many copies of the evolved genomic region is carried). The focal
males carry alleles from either the control or the population undergoing
female-limited evolution of a small genomic region. The R,P,S, and O
columns are percentages, with the last column giving the total number of
females tested.</p>
<pre class="r"><code>df &lt;- tibble(
  replicate = c(1,2,1,2),
  treatment = c(&quot;control&quot;, &quot;control&quot;, &quot;experimental&quot;, &quot;experimental&quot;),
  R = c(42, 36, 40, 17),
  P = c(26, 21, 20, 40),
  S = c(22, 33, 22, 26),
  O = c(10, 9, 17, 17),
  total = c(90, 75, 90, 70)
)

df %&gt;% 
  kable() %&gt;% 
  kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
replicate
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
R
</th>
<th style="text-align:right;">
P
</th>
<th style="text-align:right;">
S
</th>
<th style="text-align:right;">
O
</th>
<th style="text-align:right;">
total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
90
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
experimental
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
90
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
experimental
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
70
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code># first convert the 4 percentage columns to counts of females (with rounding)
df$R &lt;- round((df$R/100) * df$total)
df$P &lt;- round((df$P/100) * df$total)
df$S &lt;- round((df$S/100) * df$total)
df$O &lt;- round((df$O/100) * df$total)

# Pool the females over the 3 types representing focal males
focal_male_chosen &lt;- rowSums(df %&gt;% select(R, P, S))

# Make a 2x2 contingency table giving number of times the focal vs non-focal male mated, for control vs experimental
x &lt;- matrix(c(focal_male_chosen[1] + focal_male_chosen[2], 
         focal_male_chosen[3] + focal_male_chosen[4], 
         df$O[1] + df$O[2], 
         df$O[3] + df$O[4]), ncol = 2)

get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1992) Science 256, 1436-1439&quot;, 
           Trait = &quot;Male fitness&quot;, 
           Generations = &quot;29&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;A positive effect size means that males carrying a genomic region that underwent female-limited evolution had lower fitness compared to males carrying control genomes.&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice92.csv&quot;)</code></pre>
</div>
</div>
<div id="rice-1996" class="section level2">
<h2>Rice (1996)</h2>
<p>Rice, W. R. (1996). Sexually antagonistic male adaptation triggered
by experimental arrest of female evolution. <em>Nature</em>, 381,
232–234.</p>
<div id="male-fitness" class="section level3">
<h3>Male fitness</h3>
<div id="rices-original-data-1" class="section level4">
<h4>Rice’s original data</h4>
<p>The means for the male fitness were recorded from Figure 1, using
PlotDigitizer.</p>
<pre class="r"><code># These data come from using PlotDigitizer web app on data in the first 2 columns of Figure 2
rice96 &lt;- tibble(generation = c(rep(31,4), rep(32, 4)),
       replicate = rep(c(&quot;A&quot;, &quot;A&quot;, &quot;B&quot;, &quot;B&quot;), 2),
       treatment = rep(c(&quot;ML&quot;, &quot;Control&quot;, &quot;ML&quot;, &quot;Control&quot;), 2),
       fitness = c(1.312563038, 1.040288341, 1.240486316, 0.959122165,
                   1.233782098, 1.018216389, 1.182082107, 0.982122656),
       sample_size = rep(c(4,8), each=4)) 

rice96 %&gt;% 
  mutate(fitness = round(fitness,2)) %&gt;% 
  rename(`Male fitness` = fitness) %&gt;% 
  kable() %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
generation
</th>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
Male fitness
</th>
<th style="text-align:right;">
sample_size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
1.24
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:right;">
32
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
1.23
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:right;">
32
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:right;">
32
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
1.18
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:right;">
32
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
0.98
</td>
<td style="text-align:right;">
8
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-1" class="section level4">
<h4>Calculating effect size</h4>
<p>Unfortunately the study provides means but no standard error or
standard deviation, which we need to compute effect size variance and
confidence limits.</p>
<p>To get an estimate of effect size, we assume here the standard
deviation (SD) is the same as the SD in similar male fitness assays
carried out in Wong &amp; Holman (2023; <em>Evolution</em>). That
study’s male fitness assay methods were quite similar, and focused on
North American <em>D. melanogaster</em> as well. We load that study’s
raw data, and find the average standard deviation in proportion of
offspring sired relative to competitor males. We then use simulation to
generate 10,000 possible datasets for Rice (1996) by generating random
Normally distributed numbers using the means from Rice’s Figure 1 and SD
for Wong &amp; Holman, and from these random datasets calculate 10,000
estimates of the standardised effect size Cohen’s d comparing the two
treatments. From these 10,000 estimates we also obtain the effect size
variance and 95% CIs.</p>
<pre class="r"><code># Rice&#39;s description of the male fitness trait:
# Extra vials were constructed by mixing EA, EB, CA&#39;, and CB&#39; males with target females (4 and 8 vials each in generations 31 and 32, respectively) using a procedure matching that of the adapting-male compartment of Fig. 1, except that 75% of the males were replaced by competitor males that were homozygous for the marker pP (previously recombined into the genetic background of the LH source stock). Offspring from competitor males were pink-eyed and could be distinguished from red-eyed males carrying the genomic haplotype of the experimental (EA or EB) or control (CA&#39; or CB&#39;) males. The ratio of red-eyed to pink-eyed adult male offspring, after adjusting for Mendelian segregation, measured net relative fitness.

# Find the standard deviation in proportion offspring sired in a different study, Wong &amp; Holman 2023 Evolution, for 810 male fitness assays covering 125 fly genotypes
# Then take the average SD across those 125 genotypes, and assume this is representative of the SD in Rice&#39;s study
estimated_SD &lt;- read_csv(
  &quot;input_data/male_fitness_raw_data_wong_and_holman.csv&quot;, 
  show_col_types = FALSE) %&gt;% 
  mutate(fitness = `Early male focal` / 
           (`Early male focal` + `Early male rival`)) %&gt;% 
  group_by(Line) %&gt;% 
  summarise(mean_fitness = mean(fitness),
            sd_fitness = sd(fitness)) %&gt;% 
  summarise(x = mean(sd_fitness, na.rm=T)) %&gt;% pull(x)


# make n random normal fitness values given mean and SD
make_data &lt;- function(mean_fitness, n, sd){
  rnorm(n, mean_fitness, sd)
}

bootstrapped_effect_sizes &lt;- lapply(1:1000, function(i){
  
  possible_dataset &lt;- map_df(1:8, ~ tibble(
    generation = rice96$generation[.x],
    replicate = rice96$replicate[.x],
    treatment = rice96$treatment[.x],
    fitness = rnorm(n = rice96$sample_size[.x], 
                    mean = rice96$fitness[.x], 
                    sd = estimated_SD))) %&gt;% 
    mutate(line = paste(treatment, replicate, sep = &quot;_&quot;),
           fitness = as.numeric(scale(fitness))) # scale fitness
  
  # do a linear model (effect size of treatment is equal to Cohen&#39;s d)
  as.numeric(summary(lm(fitness ~ treatment + line, data = possible_dataset))$coefficients[2,1:2])
}) %&gt;% 
  do.call(&quot;rbind&quot;, .) %&gt;% 
  as.data.frame() %&gt;% as_tibble() %&gt;% 
  mutate(V2 = V2 ^ 2) %&gt;% # Square the effect size SE to get the variance
  rename(Estimate = V1, Var = V2)

bootstrapped_effect_sizes %&gt;% 
  summarise(Estimate = mean(Estimate), # bootstrapped Cohen&#39;s d effect size
            Var = mean(Var)) %&gt;%       # bootstrapped variance in Cohen&#39;s d effect size
  mutate(Lower_95_CI = Estimate - 1.96 * sqrt(Var),
         Upper_95_CI = Estimate + 1.96 * sqrt(Var)) %&gt;% 
    mutate(Study = &quot;Rice (1996) Nature 381, 232–234&quot;, 
           Trait = &quot;Male fitness&quot;, 
           Generations = &quot;32&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;The positive effect size means that males carrying a genomic region that underwent male-limited evolution had higher fitness than males carrying control genomes&quot;,
           .before = 1)  %&gt;% 
  write_csv(&quot;effect_sizes/rice96.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="rice-1998" class="section level2">
<h2>Rice (1998)</h2>
<p>Rice, W. R. (1998). Male fitness increases when females are
eliminated from the gene pool: Implications for the Y chromosome.
<em>PNAS</em>, 95, 6217–6221.</p>
<div id="mating-speed" class="section level3">
<h3>Mating speed</h3>
<p>Data obtained from Figure 1A, using PlotDigitiser. Rice placed males
with male-evolved or control genomes in vials with virgin females, and
counted the number of mating pairs after 15 minutes. This is a rough
measure of how quickly males induce virgin females to mate and
presumably correlates with males’ ability to find/court females,
attractiveness and mating duration.</p>
<div id="rices-original-data-2" class="section level4">
<h4>Rice’s original data</h4>
<pre class="r"><code>x &lt;- matrix(c(52, 3, 42, 13), nrow = 2)
colnames(x) &lt;- c(&quot;male-limited&quot;, &quot;control&quot;)
rownames(x) &lt;- c(&quot;mated&quot;, &quot;unmated&quot;)
kable(x) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
male-limited
</th>
<th style="text-align:right;">
control
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mated
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
42
</td>
</tr>
<tr>
<td style="text-align:left;">
unmated
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
13
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-2" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Male mating speed&quot;, 
           Generations = &quot;38&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;The positive effect size means that males carrying a haplotype that underwent male-limited evolution mated with previously-unmated females faster than males carrying control haplotypes&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_1.csv&quot;)</code></pre>
</div>
</div>
<div id="male-attractiveness-to-mated-females" class="section level3">
<h3>Male attractiveness to mated females</h3>
<p>This data is from Figure 1B (early measure), and shows the number of
previously mated female flies mating vs not-mating with either EAB or
CAB males. More females remating implies that the male type is better
able to achieve matings with mated females. Females remated more with
EAB males than CAB males.</p>
<div id="rices-raw-data" class="section level4">
<h4>Rice’s raw data</h4>
<pre class="r"><code>x &lt;- matrix(c(54, 25, 140-54, 140-25), nrow=2)
colnames(x) &lt;- c(&quot;n_remating&quot;, &quot;n_not_remating&quot;)
rownames(x) &lt;- c(&quot;EAB&quot;, &quot;CAB&quot;)
kable(x) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
n_remating
</th>
<th style="text-align:right;">
n_not_remating
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
115
</td>
</tr>
</tbody>
</table>
</div>
<div id="effect-size" class="section level4">
<h4>Effect size</h4>
<pre class="r"><code>get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Male attractiveness to mated females&quot;, 
           Generations = &quot;38&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;The positive effect size means that males carrying a haplotype that underwent male-limited evolution attained more matings with previously-mated females compared to males carrying control haplotypes&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_2.csv&quot;)</code></pre>
</div>
</div>
<div id="sperm-defence" class="section level3">
<h3>Sperm defence</h3>
<p>Data collected from Figure 1C using PlotDigitiser. The data shows the
proportion sired by the focal male (which carried either EAB or CAB
genomes) after the female mated to a second male of a standard genoptype
carrying marker mutations. Higher values indicate better ‘sperm
defence’, i.e. reduced paternity loss to the second male.</p>
<div id="rices-raw-data-1" class="section level4">
<h4>Rice’s raw data</h4>
<pre class="r"><code>m &lt;- as.data.frame(matrix(
  c(
    0.8087228560912771, 13.078892347721043,
    1.1769782848730217, 15.50243925103378,
    1.826647037173353, 23.728290665799378,
    2.1877806404122193, 19.96520001858477,
    2.8175193227824806, 19.078474190400968,
    3.191001104158999, 19.057008781303722,
    3.830622009569378, 49.462760767550996,
    4.195491350754509, 49.75561027737769),
  ncol = 2,
  byrow = TRUE)) %&gt;% 
  mutate(treatment = rep(c(&quot;EAB&quot;, &quot;CAB&quot;), 4), 
         type = rep(c(&quot;all_pink&quot;, &quot;most_pink&quot;, &quot;most_red&quot;, &quot;all_red&quot;), each=2),
         n = round(V2)) %&gt;% select(treatment, type, n)

kable(m) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:left;">
all_pink
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:left;">
all_pink
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:left;">
most_pink
</td>
<td style="text-align:right;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:left;">
most_pink
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:left;">
most_red
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:left;">
most_red
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:left;">
all_red
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:left;">
all_red
</td>
<td style="text-align:right;">
50
</td>
</tr>
</tbody>
</table>
</div>
<div id="effect-size-1" class="section level4">
<h4>Effect size</h4>
<p>To get a Cohen’s d effect size, I first reduce the results from a 4x2
contingency table to a 2x2 one, by pooling the all-red and mostly-red
categories, and the all-pink and mostly-pink categories. A positive
effect size here would mean better sperm defence in EAB males than CAB
(though the effect size does not differ significantly from zero).</p>
<pre class="r"><code>x &lt;- matrix(c(m$n[1] + m$n[3],
         m$n[2] + m$n[4],
         m$n[5] + m$n[7],
         m$n[6] + m$n[8]), nrow = 2)

get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Male sperm defence&quot;, 
           Generations = &quot;40&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;A positive effect size means that males carrying a haplotype that underwent male-limited evolution fertilise more offspring when mating first in twice-mated females, compared to males carrying control haplotypes (though there is no significant difference between the two treatments)&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_3.csv&quot;)</code></pre>
</div>
</div>
<div id="sperm-offence" class="section level3">
<h3>Sperm offence</h3>
<p>Data collected from Figure 1D using PlotDigitiser. Rice mated females
to a pink-eyed male first, then put them with a tester EAB or CAB male,
and recorded the progeny genotypes qualitatively (mostly- or all-red vs
pink). He reports a significant difference using a “directed <span
class="math inline">\(\chi^2\)</span> contingency test” (perhaps meaning
a 1-tail test), but the result appears non-significant with a standard
2-tailed test.</p>
<div id="rices-raw-data-2" class="section level4">
<h4>Rice’s raw data</h4>
<pre class="r"><code>m &lt;- as.data.frame(matrix(
  c(
    0.8123905550128108, 10.75088339222615,
    1.1750935466627344, 16.231552691748078,
    1.8095150319809772, 2.1388484722510914,
    2.169287202079224, 0.6308459779671587,
    2.7968885366168363, 11.076699230929123,
    3.1919042966949918, 21.357825815838705,
    3.8175517501981533, 76.27520266056953,
    4.176346979779175, 61.42434005404282
  ),
  ncol = 2,
  byrow = TRUE
)) %&gt;% 
  mutate(treatment = rep(c(&quot;EAB&quot;, &quot;CAB&quot;), 4), 
         type = rep(c(&quot;all_pink&quot;, &quot;most_pink&quot;, &quot;most_red&quot;, &quot;all_red&quot;), each=2),
         n = round(V2)) %&gt;% select(treatment, type, n)

m %&gt;% 
  pivot_wider(names_from = type, values_from = n) %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
all_pink
</th>
<th style="text-align:right;">
most_pink
</th>
<th style="text-align:right;">
most_red
</th>
<th style="text-align:right;">
all_red
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
76
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
61
</td>
</tr>
</tbody>
</table>
</div>
<div id="effect-size-2" class="section level4">
<h4>Effect size</h4>
<p>To get a Cohen’s d effect size, I first reduce the results from a 4x2
contingency table to a 2x2 one, by pooling the all-red and mostly-red
categories, and the all-pink and mostly-pink categories. A positive
effect size here would mean better sperm offence in EAB males than CAB
(though the effect size does not differ significantly from zero).</p>
<pre class="r"><code>x &lt;- matrix(c(m$n[5] + m$n[7],
         m$n[6] + m$n[8],
         m$n[1] + m$n[3],
         m$n[2] + m$n[4]), nrow = 2)

# positive effect means higher sperm offence for EAB males. It is not significant, contra Rice&#39;s analysis
get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Male sperm offence&quot;, 
           Generations = &quot;40&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;A positive effect size means that males carrying a haplotype that underwent male-limited evolution fertilise more offspring when mating second in twice-mated females, compared to males carrying control haplotypes (though there is no significant difference between the two treatments)&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_4.csv&quot;)</code></pre>
</div>
</div>
<div id="sex-specific-development-time" class="section level3">
<h3>Sex-specific development time</h3>
<div id="rices-original-data-3" class="section level4">
<h4>Rice’s original data</h4>
<p>These data were extracted from Figure 2A using PlotDigitizer. The
data show the numbers of females eclosed in each of the 4 replicate
lines over a few days. We do not know what fraction of eggs hatched, so
this data tells us only the development time of females that eventually
hatched. It’s also unclear if more females would have hatched on later
days.</p>
<pre class="r"><code>dt &lt;- read_csv(&quot;input_data/rice98_development_time.csv&quot;, 
               show_col_types = F) %&gt;% 
  mutate(n_females_emerged = round(prop_emerged * overall_eclosed),
         n_females_not_emerged = overall_eclosed - n_females_emerged) %&gt;% 
  mutate(treatment = substr(population, 1, 1),
         line = population, .before = 1,
         replicate = paste(population, generation, sep = &quot;_&quot;), 
         day_centered = as.numeric(scale(day))) # mean-centre day and scale by dividing by its SD 

kable(dt) %&gt;% kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
line
</th>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:right;">
day_centered
</th>
<th style="text-align:right;">
day
</th>
<th style="text-align:right;">
prop_emerged
</th>
<th style="text-align:right;">
overall_eclosed
</th>
<th style="text-align:left;">
population
</th>
<th style="text-align:right;">
generation
</th>
<th style="text-align:right;">
n_females_emerged
</th>
<th style="text-align:right;">
n_females_not_emerged
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.5311097
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
61
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.5926928
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.7550655
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
86
</td>
<td style="text-align:right;">
28
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.9427559
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
0.5266272
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
0.9972443
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_29
</td>
<td style="text-align:right;">
1.4294167
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.4769470
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.5371122
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:right;">
58
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.7500050
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
95
</td>
<td style="text-align:right;">
31
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.8978175
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
113
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
0.5266272
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
0.9863767
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_29
</td>
<td style="text-align:right;">
1.4294167
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.2708133
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
92
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.3939294
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:right;">
65
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.7496843
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.9169923
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
0.5266272
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
0.9739658
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_29
</td>
<td style="text-align:right;">
1.4294167
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.1981622
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
83
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.3488406
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
67
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.6018468
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.7704475
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
79
</td>
<td style="text-align:right;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
0.5266272
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
0.9747725
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
100
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_29
</td>
<td style="text-align:right;">
1.4294167
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.3810871
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.5640077
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.7751702
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
109
</td>
<td style="text-align:right;">
32
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.9135555
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
0.9780220
</td>
<td style="text-align:right;">
12.5
</td>
<td style="text-align:right;">
0.9889588
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
139
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:left;">
CA_30
</td>
<td style="text-align:right;">
1.8808115
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:left;">
CA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
141
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.4061285
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:right;">
88
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.5808272
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
86
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.7250661
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.8855230
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
131
</td>
<td style="text-align:right;">
17
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
0.9780220
</td>
<td style="text-align:right;">
12.5
</td>
<td style="text-align:right;">
0.9622366
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
142
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:left;">
CB_30
</td>
<td style="text-align:right;">
1.8808115
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
CB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.2168442
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
112
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.3997003
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.6844457
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
45
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.8648744
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
0.9780220
</td>
<td style="text-align:right;">
12.5
</td>
<td style="text-align:right;">
0.9621721
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
138
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:left;">
EA_30
</td>
<td style="text-align:right;">
1.8808115
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:left;">
EA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
-1.2789518
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
0.2600960
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
113
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
-0.8275571
</td>
<td style="text-align:right;">
10.5
</td>
<td style="text-align:right;">
0.4234797
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
88
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
-0.3761623
</td>
<td style="text-align:right;">
11.0
</td>
<td style="text-align:right;">
0.6160398
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
94
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
0.0752325
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
0.7607297
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
116
</td>
<td style="text-align:right;">
37
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
0.9780220
</td>
<td style="text-align:right;">
12.5
</td>
<td style="text-align:right;">
0.8687464
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
133
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:left;">
EB_30
</td>
<td style="text-align:right;">
1.8808115
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:left;">
EB
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
153
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-3" class="section level4">
<h4>Calculating effect size</h4>
<p>Here, I fit a binomial GLMM using the lme4 package with the form</p>
<p><span class="math display">\[Y = day \times treatment + (1 |
line)\]</span></p>
<p>With Y being a 2-column response variable giving the number of
females eclosed or not yet eclosed, day in standard units, treatment
being male-limited or control, and line being a 4-level factor (2
replicate lines each of the 2 treatments).</p>
<p>I then used the fitted model to estimate the expected day on which
50% of the females had eclosed for each treatment, and used parametric
bootstrapping to find the difference in number of days between the
treatments (i.e. Cohen’s d since day was standardised).</p>
<p>The positive effect size indicates <strong>slower</strong> emergence
of females carrying male-limited genomes (i.e. masculinisation of the
phenotype). Rice interprets this result as extra female-harming
mutations accumulating under male-limited selection, which is an
additional valid interpretation.</p>
<pre class="r"><code>resp &lt;- with(dt, cbind(n_females_emerged, n_females_not_emerged))
model &lt;- glmer(
  resp ~ day_centered + treatment + (1 | line),
  family = binomial,
  data = dt,
  control = glmerControl(optimizer = &quot;bobyqa&quot;)
)

# Get the log odds ratio effect size (+ SE) of treatment on Prob(eclosion) at the average value of day (i.e. day = 11.42)
LOR &lt;- coef(summary(model))[2,1:2] %&gt;% as.numeric()


tibble(
  Estimate = LOR[1] * sqrt(3) / pi,
  Var = (LOR[2]^2) * sqrt(3) / pi,
  Lower_95_CI = (LOR[1] - LOR[2]*1.96) * sqrt(3) / pi,
  Upper_95_CI = (LOR[1] + LOR[2]*1.96) * sqrt(3) / pi) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Female development time&quot;, 
           Generations = &quot;30&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;The positive effect size means females carrying haplotypes that underwent male-limited evolution took longer to complete development than females carrying control genomes.&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_5.csv&quot;)</code></pre>
</div>
</div>
<div id="female-productivity" class="section level3">
<h3>Female productivity</h3>
<p>The data are from Figure 2B (captured with PlotDigitizer). Rice
counted the “total number of adult offspring (measures net female
fitness independent of development time) produced by EAB or CAB females
(red-eyed offspring) in competition with tester-females (pink-eyed
offspring).” There was no difference, implying that EAB and CAB are
equally good at producing eggs and their offspring have equal survival
rates (or if there are differences in these fitness components, then
they cancel out). Note that in the methods and the figure axis label,
Rice mentions that he only counted the daughters, while in the figure
caption (quoted here) he says that all offspring were counted (meaning,
sons too). It’s unclear which was measured, but it likely does not
qualitatively change the result.</p>
<div id="rices-original-data-4" class="section level4">
<h4>Rice’s original data</h4>
<pre class="r"><code>x &lt;- matrix(c(349, 344, 197, 196), nrow = 2)
colnames(x) &lt;- c(&quot;focal_female_offspring&quot;, &quot;rival_female_offspring&quot;)
rownames(x) &lt;- c(&quot;CAB&quot;, &quot;EAB&quot;)

kable(x) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
focal_female_offspring
</th>
<th style="text-align:right;">
rival_female_offspring
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
349
</td>
<td style="text-align:right;">
197
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
344
</td>
<td style="text-align:right;">
196
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-4" class="section level4">
<h4>Calculating effect size</h4>
<p>Here, we define a positive effect to mean that EAB females are
<em>less</em> productive (which is what we’d predict if male-limited
selection harms female fitness), though the observed effect size is zero
or close to it.</p>
<pre class="r"><code>get_effect_size_2x2_matrix(x) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Female productivity&quot;, 
           Generations = &quot;30&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;A positive effect size means that females carrying a haplotype that underwent male-limited evolution produced fewer offspring than females carrying a control haplotype (though there is no significant difference).&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_6.csv&quot;)</code></pre>
</div>
</div>
<div id="mortality-of-males-mates" class="section level3">
<h3>Mortality of males’ mates</h3>
<div id="rices-original-data-5" class="section level4">
<h4>Rice’s original data</h4>
<p>Data extracted from Figure 3A with PlotDigitizer.</p>
<pre class="r"><code>mortality &lt;- tibble(
  treatment = rep(c(&quot;EAB&quot;, &quot;CAB&quot;), each = 7),
  day = c(
    0,1,2,3,4,5,12, 0,1,2,3,4,5,12),
  n_dead_females = c(
    0,4,7,8,9,15,29,0,0,2,2,2,5,16),
  n_live_females = 140 - n_dead_females)

kable(mortality) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
day
</th>
<th style="text-align:right;">
n_dead_females
</th>
<th style="text-align:right;">
n_live_females
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
136
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
133
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
132
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
131
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
125
</td>
</tr>
<tr>
<td style="text-align:left;">
EAB
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
111
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
138
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
138
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
138
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
135
</td>
</tr>
<tr>
<td style="text-align:left;">
CAB
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
124
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-5" class="section level4">
<h4>Calculating effect size</h4>
<p>To calculate the effect size, I fit a Cox’s proportional hazard model
to the female surivivorship data, and measured the difference between
the male-limited and control treatments in the per-day risk of death
(conditional on being alive), taking into account the right censoring of
the data (since the experiment ended on day 15, when most females were
still alive). There is a weakly significant difference, such that
females die faster when mated to EAB males, which is qualitatively the
same conclusion Rice reached (Rice used several pairwise t-tests to test
specific days one-at-a-time, which is a weaker approach, but
understandable given the statistical software available in 1998).</p>
<p>The positive effect size means that females die faster when mated to
males carrying male-evolved compared to control genomes.</p>
<pre class="r"><code>make_surv_data &lt;- function(day, cum_dead, treatment) {
  new_deaths &lt;- c(cum_dead[1], diff(cum_dead))
  death_times &lt;- rep(day, new_deaths)
  data.frame(
    time = death_times,
    event = 1,
    treatment = treatment
  )
}

surv_A &lt;- make_surv_data(
  day = mortality$day[mortality$treatment == &quot;EAB&quot;],
  cum_dead = mortality$n_dead_females[mortality$treatment == &quot;EAB&quot;],
  treatment = &quot;EAB&quot;
)

surv_B &lt;- make_surv_data(
  day = mortality$day[mortality$treatment == &quot;CAB&quot;],
  cum_dead = mortality$n_dead_females[mortality$treatment == &quot;CAB&quot;],
  treatment = &quot;CAB&quot;
)

surv_data &lt;- bind_rows(surv_A, surv_B)

censored_A &lt;- data.frame(
  time = 12,
  event = 0,
  treatment = &quot;EAB&quot;
)[rep(1, 140 - nrow(surv_A)), ]

censored_B &lt;- data.frame(
  time = 12,
  event = 0,
  treatment = &quot;CAB&quot;
)[rep(1, 140 - nrow(surv_B)), ]

surv_data &lt;- bind_rows(surv_data, censored_A, censored_B)

cox &lt;- coxph(
  Surv(time, event) ~ treatment,
  data = surv_data
)

logHR &lt;- coef(cox)[&quot;treatmentEAB&quot;]
Var_HR &lt;- vcov(cox)[&quot;treatmentEAB&quot;, &quot;treatmentEAB&quot;]
CI_logHR &lt;- logHR + c(-1, 1) * 1.96 * sqrt(Var_HR)

d &lt;- logHR * sqrt(3) / pi
Var_d &lt;- Var_HR * sqrt(3) / pi
CI_d &lt;- CI_logHR * sqrt(3) / pi

tibble(
  Estimate = d,
  Var = Var_d,
  Lower_95_CI = CI_d[1],
  Upper_95_CI = CI_d[2]) %&gt;% 
    mutate(Study = &quot;Rice (1998) PNAS, 95, 6217–6221&quot;, 
           Trait = &quot;Male harmfulness to females&quot;, 
           Generations = &quot;30&quot;,
           Organism = &quot;Drosophila melanogaster&quot;,
           Interpretation = &quot;A positive effect size means that males carrying a haplotype that underwent male-limited evolution have a negative effect on the survival of their mates.&quot;,
           .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/rice98_7.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="prasad-et-al.-2008" class="section level2">
<h2>Prasad et al. (2008)</h2>
<p>Prasad, N. G., Bedhomme, S., Day, T., &amp; Chippindale, A. K.
(2007). An evolutionary cost of separate genders revealed by
male-limited evolution. <em>The American Naturalist</em>, 169,
29–37.</p>
<div id="male-and-female-fitness" class="section level3">
<h3>Male and female fitness</h3>
<div id="prasad-et-al.s-original-data" class="section level4">
<h4>Prasad et al.’s original data</h4>
<p>The data come from Figure 2 (transcribed with PlotDigitizer). The
same size is given as n=15 per replicate population (there were 4
replicates for the control and 4 for the male-limited treatment).</p>
<pre class="r"><code>df &lt;- data.frame(
  treatment = c(&quot;control&quot;,&quot;control&quot;, &quot;male_evolved&quot;, &quot;male_evolved&quot;),
  sex = c(&quot;female&quot;, &quot;male&quot;, &quot;female&quot;, &quot;male&quot;),
  mean = c(0.858247281, 0.73376332, 0.785461824, 0.845795256),
  SE = c(0.015168311, 0.015320071, 0.015529016, 0.015670879),
  n = rep(60, 4)
)

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
0.8582473
</td>
<td style="text-align:right;">
0.0151683
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
0.7337633
</td>
<td style="text-align:right;">
0.0153201
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
0.7854618
</td>
<td style="text-align:right;">
0.0155290
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
0.8457953
</td>
<td style="text-align:right;">
0.0156709
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-6" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code># For female fitness:
mes(m.1 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;],
    m.2 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;],
    sd.1 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    sd.2 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Female fitness (2007)&quot;,
           interpretation = &quot;This positive effect size indicates that females carrying male-evolved haplotypes were less fit than control females.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_1.csv&quot;)

# For male fitness:
mes(m.2 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;],
    m.1 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;],
    sd.2 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    sd.1 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Male fitness (2007)&quot;,
           interpretation = &quot;This positive effect size indicates that males carrying male-evolved haplotypes were fitter than control males.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_2.csv&quot;)</code></pre>
</div>
</div>
<div id="development-time" class="section level3">
<h3>Development time</h3>
<div id="prasad-et-al.s-original-data-1" class="section level4">
<h4>Prasad et al.’s original data</h4>
<p>These data (development time in hours) are presented in prose in the
Results. The sample size is not recorded, so we estimated it as n=60
flies per treatment (a feasible number of individuals on the same order
as the same study’s fitness assays). An error here would over- or
under-inflate the effect size precision, but likely not have a major
effect on the overall effect size estimate.</p>
<pre class="r"><code>df &lt;- data.frame(
  treatment = c(&quot;control&quot;,&quot;control&quot;, &quot;male_evolved&quot;, &quot;male_evolved&quot;),
  sex = c(&quot;female&quot;, &quot;male&quot;, &quot;female&quot;, &quot;male&quot;),
  mean = c(224.8, 225.9, 227.0, 229.8),
  SE = c(0.61, 0.59, 0.57, 0.54),
  n = rep(60, 4)
)

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
224.8
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
225.9
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
227.0
</td>
<td style="text-align:right;">
0.57
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
229.8
</td>
<td style="text-align:right;">
0.54
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-7" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code># For female DT:
mes(m.2 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;],
    m.1 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;],
    sd.2 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    sd.1 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Female development time&quot;,
           interpretation = &quot;This positive effect size indicates that females carrying male-evolved haplotypes developed faster than control females  (i.e. have a masculinised phenotype).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_3.csv&quot;)

# For male DT:
mes(m.2 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;],
    m.1 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;],
    sd.2 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    sd.1 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Male development time&quot;,
           interpretation = &quot;This positive effect size indicates that males carrying male-evolved haplotypes developed faster than control males  (i.e. have a masculinised phenotype).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_4.csv&quot;)</code></pre>
</div>
</div>
<div id="dry-mass" class="section level3">
<h3>Dry mass</h3>
<div id="prasad-et-al.s-original-data-2" class="section level4">
<h4>Prasad et al.’s original data</h4>
<p>These data (dry mass in micrograms) are presented in prose in the
Results. The sample size is not recorded, so we estimated it as n=60
flies per treatment (a feasible number of individuals on the same order
as the same study’s fitness assays). An error here would over- or
under-inflate the effect size precision, but likely not have a major
effect on the overall effect size estimate.</p>
<pre class="r"><code>df &lt;- data.frame(
  treatment = c(&quot;control&quot;,&quot;control&quot;, &quot;male_evolved&quot;, &quot;male_evolved&quot;),
  sex = c(&quot;female&quot;, &quot;male&quot;, &quot;female&quot;, &quot;male&quot;),
  mean = c(275.9, 217.9, 267.4, 212.5),
  SE = c(1.45, 1.25, 1.82, 1.43),
  n = rep(60, 4)
)

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
275.9
</td>
<td style="text-align:right;">
1.45
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
217.9
</td>
<td style="text-align:right;">
1.25
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:right;">
267.4
</td>
<td style="text-align:right;">
1.82
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
male_evolved
</td>
<td style="text-align:left;">
male
</td>
<td style="text-align:right;">
212.5
</td>
<td style="text-align:right;">
1.43
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-8" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code># For female dry mass:
mes(m.1 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;],
    m.2 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;],
    sd.1 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    sd.2 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;female&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Female body mass (2007)&quot;,
           interpretation = &quot;This positive effect size indicates that females carrying male-evolved haplotypes are lighter than control females (i.e. have a masculinised phenotype).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_5.csv&quot;)

# For male dry mass:
mes(m.1 = df$mean[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;],
    m.2 = df$mean[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;],
    sd.1 = df$SE[df$treatment == &quot;control&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    sd.2 = df$SE[df$treatment == &quot;male_evolved&quot; &amp; df$sex == &quot;male&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose=F) %&gt;%
  parse_es(study = &quot;Prasad et al. (2007) Am. Nat. 169, 29-37&quot;,
           generations = 25,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Male body mass (2007)&quot;,
           interpretation = &quot;This positive effect size indicates that males carrying male-evolved haplotypes are lighter than control males (i.e. have a masculinised phenotype).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/prasad2007_6.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="bedhomme-et-al.-2008" class="section level2">
<h2>Bedhomme et al. (2008)</h2>
<p>Bedhomme, S., Prasad, N. G., Jiang, P. P., &amp; Chippindale, A. K.
(2008). Reproductive behaviour evolves rapidly when intralocus sexual
conflict is removed. <em>PLOS ONE</em>, 3, e2187.</p>
<p>Note: these authors also recorded the number of matings observed (in
mixed-sex vials where either the males or the females carried either
male-evolved or control genomes, while the other sex was a standard
genotype). We did not include these data, since the authors mention that
the observed number of matings probably differs substantially from the
actual number of matings. They also measured the number of flies present
near the yeast food source in those vials, but they did not score the
sex of the flies at the yeast - this makes it unclear whether the two
genome types directly affecting feeding rate, or something else
(e.g. they might affect social behaviours in a way that causes space use
near the food). These yeast data are thus difficult to interpret and we
leave them out of the meta-analysis.</p>
<p>This leaves the data on courthsip. The authors measured the number of
observed courtship events in A) vials where the males carried either of
the 2 evolved genomes and females were a standard genotype, and B) vials
where the females carried either of the 2 evolved genomes and males were
a standard genotype. For brevity, we report these two traits as ‘male
courtship rate’ and ‘female attractivenss’ respectively, though these
names are a little simplistic and hide some of the nuance in
interpreting these data.</p>
<div id="male-courtship-rate" class="section level3">
<h3>Male courtship rate</h3>
<div id="bedhomme-et-al.s-original-data" class="section level4">
<h4>Bedhomme et al.’s original data</h4>
<p>The data come from Table 1. The means represent the number of
courtship events towards females per vial, in vials where the males
carried either MLE or control genomes (females were a standard
genotype). There were 40 vials in total, 10 vials per evolving
population, with 4 replicate populations of the MLE treatment and 4 of
the control.</p>
<pre class="r"><code># The authors were surprised to see LESS courtship by the MLE males, though they note they mate just as often as controls,
# and suggest this finding reflects increased attractiveness of MLE males so they do not need to court as much

df &lt;- tibble(
  mean = c(13.6, 22.03),
  SE = c(1.13, 1.14),
  SD = SE * sqrt(40),
  n = c(40,40),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)</code></pre>
</div>
<div id="calculating-effect-size-9" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>mes(m.1 = df$mean[1], m.2 = df$mean[2],
    sd.1 = df$SD[1], sd.2 = df$SD[2],
    n.1 = 40, n.2 = 40, verbose = F) %&gt;% 
  parse_es(study = &quot;Bedhomme et al. (2008) PLOS ONE 3, e2187&quot;,
           generations = 52,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Male courtship rate&quot;,
           interpretation = &quot;This negative effect size indicates that males carrying male-evolved haplotypes courted females less often than males carrying control haplotypes. This might be because the former males are more attractive and thus need to court less.&quot;
          ) %&gt;% 
  write_csv(&quot;effect_sizes/bedhomme2008_1.csv&quot;)</code></pre>
</div>
</div>
<div id="female-attractiveness" class="section level3">
<h3>Female attractiveness</h3>
<div id="bedhomme-et-al.s-original-data-1" class="section level4">
<h4>Bedhomme et al.’s original data</h4>
<p>The data come from Table 1. The means represent the number of
courtship events towards females per vial, in vials where the females
carried either MLE or control genomes (males were a standard genotype).
There were 40 vials in total, 10 vials per evolving population, with 4
replicate populations of the MLE treatment and 4 of the control.</p>
<pre class="r"><code># The MLE females are courted slightly less, implying they are less attractive in some way (pheromones, behaviour, smaller body size?)

df &lt;- tibble(
  mean = c(4.65, 8),
  SE = c(0.66, 1.31),
  SD = SE * sqrt(40),
  n = c(40,40),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)</code></pre>
</div>
<div id="calculating-effect-size-10" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>mes(m.2 = df$mean[1], m.1 = df$mean[2],
    sd.2 = df$SD[1], sd.1 = df$SD[2],
    n.1 = 40, n.2 = 40, verbose = F) %&gt;% 
  parse_es(study = &quot;Bedhomme et al. (2008) PLOS ONE 3, e2187&quot;,
           generations = 52,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Female attractiveness&quot;,
           interpretation = &quot;This positive effect size indicates that females carrying male-evolved haplotypes were courted less often by standard males (perhaps due to a masculinisation of the phenotype after male-limited evolution?), compared to females carrying control haplotypes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/bedhomme2008_2.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="abbott-et-al.-2010" class="section level2">
<h2>Abbott et al. (2010)</h2>
<p>Abbott, J. K., Bedhomme, S., &amp; Chippindale, A. K. (2010). Sexual
conflict in wing size and shape in <em>Drosophila melanogaster</em>.
<em>Journal of Evolutionary Biology</em>, 23(9), 1989–1997.</p>
<p>The means for each sex/trait/treatment combination come from the
Dryad supplementary material. That supplement gives the sample sizes and
means for each of the 8 replicate evolution lines (not raw data), so I
took a weighted average (weighting by sample size) across the 4 lines
within each treatment to get an overall mean for both treatments, for
three male and three female traits (body mass, wing size, and fitness).
I also got the sample sizes from this same file.</p>
<p>To get the standard deviations needed for calculating effect size, I
used PlotDigitizer to get the standard error of the mean from the error
bars in Figure 2, then multiplied these by <span
class="math inline">\(\sqrt{N}\)</span>. I then calculated Cohen’s d
from the mean, SD and N using the function
<code>compute.es::mes()</code>. Positive effect sizes mean the
male-limited evolution caused phenotypic masculinisation
(i.e. <em>smaller</em> body mass and wing size, <em>higher</em> male
fitness, and <em>lower</em> female fitness).</p>
<div id="abbott-et-al.s-original-data" class="section level3">
<h3>Abbott et al.’s original data</h3>
<pre class="r"><code>abb &lt;- read_excel(&quot;input_data/Abbottetal.JEBdryaddata.xlsx&quot;) %&gt;% 
  select(treatment = `Selection treatment`,
         sex = Sex, N, body_mass = Mass, wing_size = `Wing Size`, fitness = `Relative fitness`) %&gt;% 
  filter(!is.na(sex)) %&gt;% 
  group_by(sex, treatment) %&gt;% 
  summarise(
    body_mass = sum(body_mass * N) / sum(N),
    wing_size = sum(wing_size * N) / sum(N),
    fitness = sum(fitness * N) / sum(N),
    N = sum(N), .groups = &quot;drop&quot;) %&gt;% 
  mutate(bodymass_SE = c(0.42645353268463704 - 0.4156004674677981, # digitised from error bars in Figure S1A
                     0.3828284977515815 - 0.3764849724346434,
                     0.24649113335535175 - 0.24378064581692538,
                     0.2303028378344046 - 0.22852442773303525)) %&gt;% 
  mutate(wingsize_SE = c(2.328553386369538 - 2.3147954486437996, # digitised from error bars in Figure 2a
                     2.3005653785321405 - 2.2892625292131914,
                     2.0457272019074146 - 2.036586718915984,
                     2.021764122659994 - 2.008893793819787)) %&gt;% 
  mutate(fitness_SE = c(1.074087075863868 - 1.0575940429537967, # digitised from error bars in Figure 2c
                     0.9990512054171464 - 0.9711295362645963,
                     1.0687912935478567 - 1.0388472684583931,
                     0.9872709537285416 - 0.9570842005569136)) %&gt;% 
  mutate(bodymass_SD = bodymass_SE * sqrt(N),
         wingsize_SD = wingsize_SE * sqrt(N),
         fitness_SD = fitness_SE * sqrt(N))
  
abb %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
body_mass
</th>
<th style="text-align:right;">
wing_size
</th>
<th style="text-align:right;">
fitness
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:right;">
bodymass_SE
</th>
<th style="text-align:right;">
wingsize_SE
</th>
<th style="text-align:right;">
fitness_SE
</th>
<th style="text-align:right;">
bodymass_SD
</th>
<th style="text-align:right;">
wingsize_SD
</th>
<th style="text-align:right;">
fitness_SD
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
0.4158431
</td>
<td style="text-align:right;">
2.315583
</td>
<td style="text-align:right;">
1.0567335
</td>
<td style="text-align:right;">
277
</td>
<td style="text-align:right;">
0.0108531
</td>
<td style="text-align:right;">
0.0137579
</td>
<td style="text-align:right;">
0.0164930
</td>
<td style="text-align:right;">
0.1806310
</td>
<td style="text-align:right;">
0.2289777
</td>
<td style="text-align:right;">
0.2744988
</td>
</tr>
<tr>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
0.3768941
</td>
<td style="text-align:right;">
2.286943
</td>
<td style="text-align:right;">
0.9738012
</td>
<td style="text-align:right;">
244
</td>
<td style="text-align:right;">
0.0063435
</td>
<td style="text-align:right;">
0.0113028
</td>
<td style="text-align:right;">
0.0279217
</td>
<td style="text-align:right;">
0.0990890
</td>
<td style="text-align:right;">
0.1765562
</td>
<td style="text-align:right;">
0.4361504
</td>
</tr>
<tr>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
0.2436179
</td>
<td style="text-align:right;">
2.035618
</td>
<td style="text-align:right;">
0.9541861
</td>
<td style="text-align:right;">
229
</td>
<td style="text-align:right;">
0.0027105
</td>
<td style="text-align:right;">
0.0091405
</td>
<td style="text-align:right;">
0.0299440
</td>
<td style="text-align:right;">
0.0410171
</td>
<td style="text-align:right;">
0.1383206
</td>
<td style="text-align:right;">
0.4531353
</td>
</tr>
<tr>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
ML
</td>
<td style="text-align:right;">
0.2287680
</td>
<td style="text-align:right;">
2.010399
</td>
<td style="text-align:right;">
1.0401660
</td>
<td style="text-align:right;">
215
</td>
<td style="text-align:right;">
0.0017784
</td>
<td style="text-align:right;">
0.0128703
</td>
<td style="text-align:right;">
0.0301868
</td>
<td style="text-align:right;">
0.0260766
</td>
<td style="text-align:right;">
0.1887161
</td>
<td style="text-align:right;">
0.4426247
</td>
</tr>
</tbody>
</table>
</div>
<div id="female-body-mass-effect-size" class="section level3">
<h3>Female body mass effect size</h3>
<pre class="r"><code>mes(m.1 = abb$body_mass[1], m.2 = abb$body_mass[2],
    sd.1 = abb$bodymass_SD[1], sd.2 = abb$bodymass_SD[2],
    n.1 = abb$N[1], n.2 = abb$N[2], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Female body mass (2010)&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that females carrying male-evolved haplotypes have lower body mass than females carrying control genotypes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_1.csv&quot;)</code></pre>
</div>
<div id="male-body-mass-effect-size" class="section level3">
<h3>Male body mass effect size</h3>
<pre class="r"><code>mes(m.1 = abb$body_mass[3], m.2 = abb$body_mass[4],
    sd.1 = abb$bodymass_SD[3], sd.2 = abb$bodymass_SD[4],
    n.1 = abb$N[3], n.2 = abb$N[4], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Male body mass  (2010)&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that males carrying male-evolved haplotypes have lower body mass than males carrying control genotypes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_2.csv&quot;)</code></pre>
</div>
<div id="female-wing-size-effect-size" class="section level3">
<h3>Female wing size effect size</h3>
<pre class="r"><code>mes(m.1 = abb$wing_size[1], m.2 = abb$wing_size[2],
    sd.1 = abb$wingsize_SD[1], sd.2 = abb$wingsize_SD[2],
    n.1 = abb$N[1], n.2 = abb$N[2], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Female wing size&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would indicate that females carrying male-evolved haplotypes have smaller wings than females carrying control genotypes (though wing size does not differ consistently).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_3.csv&quot;)</code></pre>
</div>
<div id="male-wing-size-effect-size" class="section level3">
<h3>Male wing size effect size</h3>
<pre class="r"><code>mes(m.1 = abb$wing_size[3], m.2 = abb$wing_size[4],
    sd.1 = abb$wingsize_SD[3], sd.2 = abb$wingsize_SD[4],
    n.1 = abb$N[3], n.2 = abb$N[4], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Male wing size&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would indicate that males carrying male-evolved haplotypes have smaller wings than males carrying control genotypes (though wing size does not differ consistently).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_4.csv&quot;)</code></pre>
</div>
<div id="female-fitness-effect-size" class="section level3">
<h3>Female fitness effect size</h3>
<pre class="r"><code>mes(m.1 = abb$fitness[1], m.2 = abb$fitness[2],
    sd.1 = abb$fitness_SD[1], sd.2 = abb$fitness_SD[2],
    n.1 = abb$N[1], n.2 = abb$N[2], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Female fitness (2010)&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that females carrying male-evolved haplotypes have lower fitness thanthan females carrying control genotypes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_5.csv&quot;)</code></pre>
</div>
<div id="male-fitness-effect-size" class="section level3">
<h3>Male fitness effect size</h3>
<pre class="r"><code>mes(m.2 = abb$fitness[3], m.1 = abb$fitness[4],
    sd.2 = abb$fitness_SD[3], sd.1 = abb$fitness_SD[4],
    n.2 = abb$N[3], n.1 = abb$N[4], verbose=F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2010) J. Evol. Biol. 23, 1989–1997&quot;, 
           trait = &quot;Male fitness  (2010)&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that males carrying male-evolved haplotypes have higher fitness than males carrying control genotypes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott2010_6.csv&quot;)</code></pre>
</div>
</div>
<div id="jiang-et-al.-2011" class="section level2">
<h2>Jiang et al. (2011)</h2>
<p>Jiang, P. P., Bedhomme, S., Prasad, N. G., &amp; Chippindale, A. K.
(2011). Sperm competition and mate harm unresponsive to male-limited
selection in <em>Drosophila</em>: An evolving genetic architecture under
domestication. <em>Evolution</em>, 65(9), 2448–2460.</p>
<div id="mating-defence" class="section level3">
<h3>Mating defence</h3>
<div id="jiang-et-al.s-original-data" class="section level4">
<h4>Jiang et al.’s original data</h4>
<p>These data were transcribed from the right 2 columns in Figure 2A
(i.e. using the data for LH_ST females, since these better measure the
effects of male-limitation rather than adaptation to the clone generator
females) using PlotDigitiser. The sample size is estimated based on the
overall number of replicates being 12.6 (mentioned in the text) with 10
females per replicate. This is a bit pseudo-replicated, overstating the
precision of the effect size.</p>
<pre class="r"><code>df &lt;- tibble(
  mean = c(0.6349673427294603, 0.5952973530422826),
  SE = c(0.6734272946029564 - 0.6349673427294603, 
         0.628030250945342 - 0.5952973530422826),
  SD = SE * sqrt(126),
  N = c(126, 126),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)

df %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
SD
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:left;">
treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.6349673
</td>
<td style="text-align:right;">
0.0384600
</td>
<td style="text-align:right;">
0.4317119
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
MLE
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5952974
</td>
<td style="text-align:right;">
0.0327329
</td>
<td style="text-align:right;">
0.3674259
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
control
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="calculating-effect-size-11" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[2], m.2 = df$mean[1],
    sd.1 = df$SD[2], sd.2 = df$SD[1],
    n.1 = df$N[2], n.2 = df$N[1], verbose = F) %&gt;% 
  parse_es(study = &quot;Jiang et al. (2011) Evolution 65: 2448-2460&quot;, 
           trait = &quot;Mating defence&quot;, 
           generations = 65, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would mean that the mates of males carrying haplotypes that had undergone male-limited evolution are less likely to remate with a new male, compared to the mates of control males (the effect goes non-significantly in the opposite direction, though).&quot;) %&gt;% 
   write_csv(&quot;effect_sizes/jiang_2011_1.csv&quot;)</code></pre>
</div>
<div id="sperm-defence-1" class="section level3">
<h3>Sperm defence</h3>
<div id="jiang-et-al.s-original-data-1" class="section level4">
<h4>Jiang et al.’s original data</h4>
<p>These data were transcribed from the right 2 columns in Figure 2B
(i.e. using the data for LH_ST females, since these better measure the
effects of male-limitation rather than adaptation to the clone generator
females) using PlotDigitiser. Same sample size estimate as before.</p>
<pre class="r"><code>df &lt;- tibble(
  mean = c(0.293426235316184, 0.2820820973803586),
  SE = c(0.30766024550020493 - 0.293426235316184, 
         0.29350959701565116 - 0.2820820973803586),
  SD = SE * sqrt(126),
  N = c(126, 126),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)

df %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
SD
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:left;">
treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.2934262
</td>
<td style="text-align:right;">
0.0142340
</td>
<td style="text-align:right;">
0.1597764
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
MLE
</td>
</tr>
<tr>
<td style="text-align:right;">
0.2820821
</td>
<td style="text-align:right;">
0.0114275
</td>
<td style="text-align:right;">
0.1282734
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
control
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="calculating-effect-size-12" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[1], m.2 = df$mean[2],
    sd.1 = df$SD[1], sd.2 = df$SD[2],
    n.1 = df$N[1], n.2 = df$N[2], verbose = F) %&gt;% 
  parse_es(study = &quot;Jiang et al. (2011) Evolution 65: 2448-2460&quot;, 
           trait = &quot;Sperm defence&quot;, 
           generations = 65, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would mean that males carrying haplotypes that had undergone male-limited evolution sire more offspring when mating first with twice-mated females, compared to control males (the effect goes non-significantly in the opposite direction, though).&quot;) %&gt;% 
   write_csv(&quot;effect_sizes/jiang_2011_2.csv&quot;)</code></pre>
</div>
<div id="male-mating-success" class="section level3">
<h3>Male mating success</h3>
<div id="jiang-et-al.s-original-data-2" class="section level4">
<h4>Jiang et al.’s original data</h4>
<p>These data were transcribed from the right 2 columns in Figure 3A
(i.e. using the data for LH_ST females, since these better measure the
effects of male-limitation rather than adaptation to the clone generator
females) using PlotDigitiser. Same sample size estimate as before. Note
this trait is called ‘Mating offence’ in the paper since the females had
previously mated, but we’ve renamed it here for clarity outside the
context of Jiang et al. and consistency with other traits in the
meta-analysis.</p>
<pre class="r"><code>df &lt;- tibble(
  mean = c(0.7297174519489161, 0.730766769892253),
  SE = c(0.7630716863307143 - 0.7297174519489161, 
         0.7582729762239909 - 0.730766769892253),
  SD = SE * sqrt(126),
  N = c(126, 126),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)

df %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
SD
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:left;">
treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.7297175
</td>
<td style="text-align:right;">
0.0333542
</td>
<td style="text-align:right;">
0.3744004
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
MLE
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7307668
</td>
<td style="text-align:right;">
0.0275062
</td>
<td style="text-align:right;">
0.3087564
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
control
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="calculating-effect-size-13" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[1], m.2 = df$mean[2],
    sd.1 = df$SD[1], sd.2 = df$SD[2],
    n.1 = df$N[1], n.2 = df$N[2], verbose = F) %&gt;% 
  parse_es(study = &quot;Jiang et al. (2011) Evolution 65: 2448-2460&quot;, 
           trait = &quot;Male mating success&quot;, 
           generations = 65, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would mean that the males carrying haplotypes that had undergone male-limited evolution are more likely to mate with females, compared to the mates of control males (there is no difference in mating rates, though).&quot;) %&gt;% 
   write_csv(&quot;effect_sizes/jiang_2011_3.csv&quot;)</code></pre>
</div>
<div id="sperm-offence-1" class="section level3">
<h3>Sperm offence</h3>
<div id="jiang-et-al.s-original-data-3" class="section level4">
<h4>Jiang et al.’s original data</h4>
<p>These data were transcribed from the right 2 columns in Figure 3B
(i.e. using the data for LH_ST females, since these better measure the
effects of male-limitation rather than adaptation to the clone generator
females) using PlotDigitiser. Same sample size estimate as before.</p>
<pre class="r"><code>df &lt;- tibble(
  mean = c(0.8227287156371025, 0.8270903434702559),
  SE = c(0.8284807313821345 - 0.8227287156371025, 
         0.8337502380801219 - 0.8270903434702559),
  SD = SE * sqrt(126),
  N = c(126, 126),
  treatment = c(&quot;MLE&quot;, &quot;control&quot;)
)

df %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
SD
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:left;">
treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.8227287
</td>
<td style="text-align:right;">
0.0057520
</td>
<td style="text-align:right;">
0.0645662
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
MLE
</td>
</tr>
<tr>
<td style="text-align:right;">
0.8270903
</td>
<td style="text-align:right;">
0.0066599
</td>
<td style="text-align:right;">
0.0747571
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:left;">
control
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="calculating-effect-size-14" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[1], m.2 = df$mean[2],
    sd.1 = df$SD[1], sd.2 = df$SD[2],
    n.1 = df$N[1], n.2 = df$N[2], verbose = F) %&gt;% 
  parse_es(study = &quot;Jiang et al. (2011) Evolution 65: 2448-2460&quot;, 
           trait = &quot;Sperm offence&quot;, 
           generations = 65, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;Apositive effect size would mean that males carrying haplotypes that had undergone male-limited evolution sire more offspring when mating second with twice-mated females, compared to control males (there is no difference in mating rates, though).&quot;) %&gt;% 
   write_csv(&quot;effect_sizes/jiang_2011_4.csv&quot;)</code></pre>
</div>
</div>
<div id="bedhomme-et-al.-2011" class="section level2">
<h2>Bedhomme et al. (2011)</h2>
<p>Bedhomme, S., Chippindale, A. K., Prasad, N. G., Delcourt, M.,
Abbott, J. K., Mallet, M. A., &amp; Rundle, H. D. (2011). Male-limited
evolution suggests no extant intralocus sexual conflict over the
sexually dimorphic cuticular hydrocarbons of <em>Drosophila
melanogaster</em>. <em>Journal of Genetics</em>, 90(3), 443–452.</p>
<p>This paper measured the abundance of dozens of hydrocarbons present
on the body surface of both sexes, in the male-limited evolution
treatment or the control lines. The authors analysed each hydrocarbon
with its own univariate test, and largely found no differences between
treatments.</p>
<p>Since there are many hydrocarbons, we have not included them all
here: we focus on a single male-unique hydrocarbon, 7-C<span
class="math inline">\(_{23:1}\)</span>, which is known to function as a
male sex pheromone (e.g. Grillet et al., 2006; <a
href="https://doi.org/10.1098/rspb.2005.3332"
class="uri">https://doi.org/10.1098/rspb.2005.3332</a>), because this
seems the most likely one that might evolve following removal of
selection on females. The means and standard errors in the code chunk
below come from row M7 of Table 1, and the sample sizes are inferred
from the methods (30 males per selection line, 4 selection lines per
treatment = 120 males per treatment assuming no missing samples). The
positive (non-significant) effect size implies more sex pheromones in
the ML males.</p>
<p>The authors also carried out multivariate tests on the first 3
principal components, but the results for those tests are not detailed
enough to include in this meta-analysis. The multivariate tests suggest
no difference in CHCs between treatments, for either males or
females.</p>
<div id="bedhomme-et-al.s-original-data-2" class="section level3">
<h3>Bedhomme et al.’s original data</h3>
<pre class="r"><code>tibble(treatment = c(&quot;Control&quot;, &quot;Male-limited evolution&quot;),
       mean = c(34.518, 35.003),
       SE = c(0.351, 0.322),
       N = c(120, 120)) %&gt;% 
  kable() %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
N
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
34.518
</td>
<td style="text-align:right;">
0.351
</td>
<td style="text-align:right;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
Male-limited evolution
</td>
<td style="text-align:right;">
35.003
</td>
<td style="text-align:right;">
0.322
</td>
<td style="text-align:right;">
120
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-15" class="section level3">
<h3>Calculating effect size</h3>
<pre class="r"><code>mes(m.2 = 34.518, m.1 = 35.003,
    sd.2 = 0.351 * sqrt(120), sd.1 = 0.322 * sqrt(120),
    n.1 = 120, n.2 = 120, verbose = F) %&gt;% 
  parse_es(study = &quot;Bedhomme et al. (2011) J. Genetics 90, 443–452&quot;, 
           trait = &quot;Male 7-C23:1 sex pheromone&quot;, 
           generations = 82, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would imply that males carrying male-limited haplotypes produce more 7-C23:1 than males carrying control genomes (though there is no significant difference).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/bedhomme2011.csv&quot;)</code></pre>
</div>
</div>
<div id="abbott-et-al.-2013" class="section level2">
<h2>Abbott et al. (2013)</h2>
<p>Abbott, J. K., Innocenti, P., Chippindale, A. K., &amp; Morrow, E. H.
(2013). Epigenetics and sex-specific fitness: An experimental test using
male-limited evolution in <em>Drosophila melanogaster</em>. <em>PLOS
ONE</em>, 8, e70493.</p>
<p>Here we focus on the effect size comparing the CDX and MLX
treatments, since CDX is a stronger control than the other group
(labelled C).</p>
<p>Two traits are recorded here: 1) male fitness, and 2) the proportion
of males among the progeny that survived to adulthood. Both have
positive effect sizes, indicating higher values in the male-limited
(MLX) treatment relative to the control (CDX).</p>
<p>For the second trait, the excess of males among surviving progeny
implies either a negative genetic effect of male-limited evolution on
female survival to adulthood, or some sort of non-genetic effect of the
father’s genotype on offspring sex ratio - it cannot be that the
male-limited X has a positive genetic effect on male survival (since
males do not pass on the X chromosome to their sons).</p>
<div id="abbott-et-al.s-original-data-1" class="section level3">
<h3>Abbott et al.’s original data</h3>
<p>The data were extracted from Figures 3 and S4 using PlotDigitizer. I
have assumed that the standard errors were calculated using individual
vials (rather that experimental evolution lines) as replicates, given
the small error bars, for a sample of 60 per treatment (20 for each of 3
replicate lines).</p>
<pre class="r"><code>df &lt;- tibble(
  trait = c(&quot;Male fitness&quot;, &quot;Male fitness&quot;, &quot;Proportion of sons&quot;, &quot;Proportion of sons&quot;),
  treatment = c(&quot;CDX&quot;, &quot;MLX&quot;, &quot;CDX&quot;, &quot;MLX&quot;),
  mean  = c(0.7781242592641365, 1.2066580631567856, 0.3816372189736579, 0.4524972141486783),
  SE = c(0.8454581369011561 - 0.7781242592641365,
         1.2642102768047616 - 1.2066580631567856,
         0.38419101175227177 - 0.3816372189736579, 
         0.45990786586556687 - 0.4524972141486783),
  N = c(60, 60, 60, 60)) 

df %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
trait
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
N
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.7781243
</td>
<td style="text-align:right;">
0.0673339
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
1.2066581
</td>
<td style="text-align:right;">
0.0575522
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Proportion of sons
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.3816372
</td>
<td style="text-align:right;">
0.0025538
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Proportion of sons
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.4524972
</td>
<td style="text-align:right;">
0.0074107
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="male-fitness-1" class="section level3">
<h3>Male fitness</h3>
<p>Fitness was measured as the proportion of adult offspring sired when
in competition for matings with marked competitor males.</p>
<pre class="r"><code>df &lt;- df %&gt;% mutate(SD = SE * sqrt(N))

mes(m.1 = df$mean[2], m.2 = df$mean[1],
    sd.1 = df$SD[2], sd.2 = df$SD[1],
    n.1 = df$N[2], n.2 = df$N[1], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2013) PLOS ONE 8, e70493&quot;, 
           trait = &quot;Male fitness&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;Male fitness is higher in males carrying an X chromosome that underwent male-limited evolution compared to males carrying an X that experienced only 1 generation of father-to-son inheritance.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2013_1.csv&quot;)</code></pre>
<pre class="r"><code>mes(m.1 = df$mean[4], m.2 = df$mean[3],
    sd.1 = df$SD[4], sd.2 = df$SD[3],
    n.1 = df$N[4], n.2 = df$N[3], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2013) PLOS ONE 8, e70493&quot;, 
           trait = &quot;% sons among surviving progeny&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The proportion of sons among the progeny of males carrying an X chromosome that underwent male-limited evolution is higher than that of males carrying an X that experienced only 1 generation of father-to-son inheritance.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2013_2.csv&quot;)</code></pre>
</div>
</div>
<div id="abbott-et-al.-2020" class="section level2">
<h2>Abbott et al. (2020)</h2>
<p>Abbott, J. K., Chippindale, A. K., &amp; Morrow, E. H. (2020). The
microevolutionary response to male-limited X-chromosome evolution in
<em>Drosophila melanogaster</em> reflects macroevolutionary patterns.
<em>Journal of Evolutionary Biology</em>, 33, 738–750.</p>
<p>All data transcribed using PlotDigitiser from Figures 2A, 2B, S1 and
S2. Sample sizes given in the Methods, bottom left of page 741.</p>
<p>When deciding on coding the sign of the effect sizes, I considered
high locomotion a male-typical trait and so predicted both female and
male locomotion to increase in the MLX treatment. I also predicted
increased offspring survival in MLX, due to the unmasking of deleterious
alleles in males, and an elevated proportion of sons (whether the X was
present in the mother or the father) due to the removal of selection on
female-harming alleles and (in the case of the X coming from the mother)
extra purging of male-harming alleles.</p>
<div id="abbott-et-al.s-original-data-2" class="section level3">
<h3>Abbott et al.’s original data</h3>
<pre class="r"><code>df &lt;- tibble(
  trait = c(&quot;Female fitness&quot;, &quot;Female fitness&quot;,
            &quot;Male fitness&quot;, &quot;Male fitness&quot;, 
            &quot;Female locomotion&quot;, &quot;Female locomotion&quot;,
            &quot;Male locomotion&quot;, &quot;Male locomotion&quot;,
            &quot;Female effect on % sons&quot;, &quot;Female effect on % sons&quot;,
            &quot;Male effect on % sons&quot;, &quot;Male effect on % sons&quot;,
            &quot;Egg-to-adult survival&quot;, &quot;Egg-to-adult survival&quot;),
  treatment = rep(c(&quot;CDX&quot;, &quot;MLX&quot;), 7),
  mean  = c(0.9502716648321987, 1.0522634062212999, 
            0.8833110159213118, 1.117530268720336,
            0.13367807311356172, 0.33332353958957683,
            0.44689179218981584, 0.3798242349453836,
            0.18625256770849793, 0.34924785892614485,
            0.3340217425654963, 0.4002676737351073,
            0.4647589355305646,0.5514258496331762),
  SE = c(1.0482869341484446 - 0.9502716648321987,
         1.0971487461353486 - 1.0522634062212999,
         0.9376523952074818 - 0.8833110159213118, 
         1.1710079291077475 - 1.117530268720336,
         0.16792026586749717 - 0.13367807311356172,
         0.367418826187165 - 0.33332353958957683,
         0.4805985936183966 - 0.44689179218981584,
         0.3798242349453836 - 0.3454253422913443,
         0.20611478052017823 - 0.18625256770849793,
         0.3695196409948488 - 0.34924785892614485,
         0.34469898555762735 - 0.3340217425654963,
         0.4139503207660462 - 0.4002676737351073,
         0.5538655680510708 - 0.4647589355305646,
         0.6240087097452648 - 0.5514258496331762),
  N = c(rep(60, 4), rep(30, 4), rep(60, 6))) 

df %&gt;% 
  kable() %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
trait
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
N
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female fitness
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.9502717
</td>
<td style="text-align:right;">
0.0980153
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
1.0522634
</td>
<td style="text-align:right;">
0.0448853
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.8833110
</td>
<td style="text-align:right;">
0.0543414
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
1.1175303
</td>
<td style="text-align:right;">
0.0534777
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Female locomotion
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.1336781
</td>
<td style="text-align:right;">
0.0342422
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Female locomotion
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.3333235
</td>
<td style="text-align:right;">
0.0340953
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Male locomotion
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.4468918
</td>
<td style="text-align:right;">
0.0337068
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Male locomotion
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.3798242
</td>
<td style="text-align:right;">
0.0343989
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Female effect on % sons
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.1862526
</td>
<td style="text-align:right;">
0.0198622
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Female effect on % sons
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.3492479
</td>
<td style="text-align:right;">
0.0202718
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Male effect on % sons
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.3340217
</td>
<td style="text-align:right;">
0.0106772
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Male effect on % sons
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.4002677
</td>
<td style="text-align:right;">
0.0136826
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Egg-to-adult survival
</td>
<td style="text-align:left;">
CDX
</td>
<td style="text-align:right;">
0.4647589
</td>
<td style="text-align:right;">
0.0891066
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
Egg-to-adult survival
</td>
<td style="text-align:left;">
MLX
</td>
<td style="text-align:right;">
0.5514258
</td>
<td style="text-align:right;">
0.0725829
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="female-fitness-effect-size-1" class="section level3">
<h3>Female fitness effect size</h3>
<pre class="r"><code>df &lt;- df %&gt;% mutate(SD = SE * sqrt(N))

mes(m.1 = df$mean[1], m.2 = df$mean[2],
    sd.1 = df$SD[1], sd.2 = df$SD[2],
    n.1 = df$N[1], n.2 = df$N[2], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Female fitness&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would mean that females carrying a male-evolved X chromosome have lower fitness than females carrying a control genome (the effect is non-significantly negative).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_1.csv&quot;)</code></pre>
</div>
<div id="male-fitness-effect-size-1" class="section level3">
<h3>Male fitness effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[4], m.2 = df$mean[3],
    sd.1 = df$SD[4], sd.2 = df$SD[3],
    n.1 = df$N[4], n.2 = df$N[3], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Male fitness&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that males carrying a male-evolved X chromosome have higher fitness than males carrying a control genome.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_2.csv&quot;)</code></pre>
</div>
<div id="female-locomotion" class="section level3">
<h3>Female locomotion</h3>
<pre class="r"><code>mes(m.1 = df$mean[6], m.2 = df$mean[5],
    sd.1 = df$SD[6], sd.2 = df$SD[5],
    n.1 = df$N[6], n.2 = df$N[5], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Female locomotion&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that females carrying a male-evolved X chromosome have greater locomotory activity than females with a control genome.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_3.csv&quot;)</code></pre>
</div>
<div id="male-locomotion" class="section level3">
<h3>Male locomotion</h3>
<pre class="r"><code>mes(m.1 = df$mean[8], m.2 = df$mean[7],
    sd.1 = df$SD[8], sd.2 = df$SD[7],
    n.1 = df$N[8], n.2 = df$N[7], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Male locomotion&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would indicate that males carrying a male-evolved X chromosome have greater locomotory activity than males with a control genome (there is a non-significant trend in the opposite direction, though).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_4.csv&quot;)</code></pre>
</div>
<div id="female-effect-on-sons" class="section level3">
<h3>Female effect on % sons</h3>
<pre class="r"><code>mes(m.1 = df$mean[10], m.2 = df$mean[9],
    sd.1 = df$SD[10], sd.2 = df$SD[9],
    n.1 = df$N[10], n.2 = df$N[9], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Female effect on % sons&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that females carrying a male-evolved X chromosome have a greater proportion of sons among their progeny that reach adulthood (indicating enhanced male survival, reduced female survival, and/or changes to the primary sex ratio). Note that all sons and 50% daughters actually inherit the male-evolved X from the tester females.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_5.csv&quot;)</code></pre>
</div>
<div id="male-effect-on-sons" class="section level3">
<h3>Male effect on % sons</h3>
<pre class="r"><code>mes(m.1 = df$mean[12], m.2 = df$mean[11],
    sd.1 = df$SD[12], sd.2 = df$SD[11],
    n.1 = df$N[12], n.2 = df$N[11], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Male effect on % sons&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect size indicates that males carrying a male-evolved X chromosome have a greater proportion of sons among their progeny that reach adulthood (indicating enhanced male survival, reduced female survival, and/or changes to the primary sex ratio). Note that all daughters and none of the sons actually inherit the male-evolved X from the tester males.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_6.csv&quot;)</code></pre>
</div>
<div id="egg-to-adult-survival" class="section level3">
<h3>Egg-to-adult survival</h3>
<pre class="r"><code>mes(m.1 = df$mean[14], m.2 = df$mean[13],
    sd.1 = df$SD[14], sd.2 = df$SD[13],
    n.1 = df$N[14], n.2 = df$N[13], verbose = F) %&gt;% 
  parse_es(study = &quot;Abbott et al. (2020) J. Evol. Biol. 33, 738–750&quot;, 
           trait = &quot;Egg-to-adult survival&quot;, 
           generations = 40, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;A positive effect size would indicates elevated egg-to-adult survival in the progeny of females carrying a male-evolved X chromosome relative to females carrying control genomes. Note that all sons and 50% daughters actually inherit the male-evolved X from the tester females. There is seemingly no difference based on the available data, though the authors find that MLX has greater survival than the control in a quasibinomial GLM presented in the supplement.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/abbott_2020_5.csv&quot;)</code></pre>
</div>
</div>
<div id="lund-hansen-et-al.-2020" class="section level2">
<h2>Lund-Hansen et al. (2020)</h2>
<p>Lund-Hansen, K. K., Abbott, J. K., &amp; Morrow, E. H. (2020).
Feminization of complex traits in <em>Drosophila melanogaster</em> via
female-limited X chromosome evolution. <em>Evolution</em>, 74,
2703–2713.</p>
<p>Here, I focus on comparisons between the female-limited X chromosome
treatment and the ‘control FM’ treatment (ignoring the wild-type
control), since differences between the former 2 treatments are expected
to arise due to female-limited inheritance of the X (rather than
adaption to the FM balancer chromosome). I also focus on the later
measurements made around generations 39-41, rather than the early
measurements, since the late measurements have given more time for
experimental evolution to produce evolution.</p>
<p>Effect sizes are calculated from models that motly follow those in
the authors’ archived code, with a couple of differences. First, I did
not run one model for both sexes (with sex as a predictor) as in the
original, but rather split the data by sex and ran two models (running
with both sexes together gave warnings about rank deficiency). Second, I
treated the experimental replicate lines as fixed rather than random
factors (since there are only 3 replicates per treatment, and it’s
recommended to have more levels in order to use random factors). The
results are very similar to those in the paper. Third, for the
development time data, the response variable (time until development)
was not normal, as is common for time-to-event variables, and so I used
a Cox PH model instead of linear (mixed) model as in the original paper.
The treatment effect (as a proportional hazard ratio) was converted to
Cohen’s d effect size.</p>
<div id="female-fitness-data" class="section level3">
<h3>Female fitness data</h3>
<p>These are for generation 39-41, from the file
StandardFitnessG3941.csv. I ran a model following that used by the
authors, as shown in their archived code.</p>
<pre class="r"><code>SF.data &lt;- read_csv(&quot;input_data/lundhansen_github_data_FLXevoexp-master/StandardFitnessGeneration39-41 /StandardFitnessG3941.csv&quot;, show_col_types = F)
female_SF &lt;- SF.data[SF.data$sex == &quot;f&quot;, ] %&gt;% 
  mutate(standard = as.numeric(scale(standard)))
male_SF &lt;- SF.data[SF.data$sex == &quot;m&quot;, ] %&gt;% 
  mutate(standard = as.numeric(scale(standard)))

# control FM populations have non-sig higher female fitness than FLX populations (i.e. unexpected direction)
lund_females &lt;- lm(standard ~ regime + rep_population, data = female_SF)

coef &lt;- coefficients(summary(lund_females))

effects &lt;- tibble(Estimate = coef[2,1],
                  Var = coef[2,2] ^ 2,
                  Lower_95_CI = coef[2,1] - coef[2,2]*1.96,
                  Upper_95_CI = coef[2,1] + coef[2,2]*1.96)


tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Female fitness&quot;,
       Generations = 41,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that females carrying female-evolved X chromosomes have higher fitness than the control FM genotype (there is no difference, though).&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_1.csv&quot;)</code></pre>
</div>
<div id="male-fitness-2" class="section level3">
<h3>Male fitness</h3>
<pre class="r"><code># control FM populations have higher male fitness than FLX populations 
lund_males &lt;- lm(standard ~ regime + rep_population, data = male_SF)

coef &lt;- coefficients(summary(lund_males))

effects &lt;- tibble(Estimate = coef[2,1],
                  Var = coef[2,2] ^ 2,
                  Lower_95_CI = coef[2,1] - coef[2,2]*1.96,
                  Upper_95_CI = coef[2,1] + coef[2,2]*1.96)


tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Male fitness&quot;,
       Generations = 41,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;The positive effect indicates that males carrying female-evolved X chromosomes have lower fitness than the control FM genotype (there is no difference, though).&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_2.csv&quot;)</code></pre>
</div>
<div id="female-development-time" class="section level3">
<h3>Female development time</h3>
<pre class="r"><code>dev_time &lt;- read_csv(&quot;input_data/lundhansen_github_data_FLXevoexp-master/DevelopmentTimeGeneration/DevelopmentTimeG43.csv&quot;, show_col_types = F)
dev_time_females &lt;- dev_time %&gt;% 
  filter(sex == &quot;f&quot;) %&gt;% 
  mutate(dev_time = as.numeric(scale(dev_time)))
dev_time_males &lt;- dev_time %&gt;% 
  filter(sex == &quot;m&quot;) %&gt;% 
  mutate(dev_time = as.numeric(scale(dev_time)))

# CFM females take non-sig longer to develop than FLX (this is the prediction, since shorter development is female-typical)
surv_obj &lt;- Surv(dev_time_females$dev_time, event = rep(1, nrow(dev_time_females)))
cox &lt;- coxph(surv_obj ~ regime, data = dev_time_females)

logHR &lt;- coef(cox)[&quot;regimebCFM&quot;]
Var_HR &lt;- vcov(cox)[&quot;regimebCFM&quot;, &quot;regimebCFM&quot;]
CI_logHR &lt;- logHR + c(-1, 1) * 1.96 * sqrt(Var_HR)

effects &lt;- tibble(Estimate = logHR * sqrt(3) / pi,
                  Var = Var_HR * sqrt(3) / pi,
                  Lower_95_CI = CI_logHR[1] * sqrt(3) / pi,
                  Upper_95_CI = CI_logHR[2] * sqrt(3) / pi)

tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Female development time&quot;,
       Generations = 43,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that females carrying female-evolved X chromosomes take less time to develop higher fitness than the control FM genotype (there is no difference, though).&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_3.csv&quot;)</code></pre>
</div>
<div id="male-development-time" class="section level3">
<h3>Male development time</h3>
<pre class="r"><code># CFM males take non-sig longer to develop than FLX (this is the prediction, since shorter development is a feminine trait)
surv_obj &lt;- Surv(dev_time_males$dev_time, event = rep(1, nrow(dev_time_males)))
cox &lt;- coxph(surv_obj ~ regime, data = dev_time_males)

logHR &lt;- coef(cox)[&quot;regimebCFM&quot;]
Var_HR &lt;- vcov(cox)[&quot;regimebCFM&quot;, &quot;regimebCFM&quot;]
CI_logHR &lt;- logHR + c(-1, 1) * 1.96 * sqrt(Var_HR)

effects &lt;- tibble(Estimate = logHR * sqrt(3) / pi,
                  Var = Var_HR * sqrt(3) / pi,
                  Lower_95_CI = CI_logHR[1] * sqrt(3) / pi,
                  Upper_95_CI = CI_logHR[2] * sqrt(3) / pi)

tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Male development time&quot;,
       Generations = 43,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that males carrying female-evolved X chromosomes take less time to develop higher fitness than the control FM genotype (there is no difference, though).&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_4.csv&quot;)</code></pre>
</div>
<div id="female-locomotion-1" class="section level3">
<h3>Female locomotion</h3>
<pre class="r"><code>loco &lt;- read_csv(&quot;input_data/lundhansen_github_data_FLXevoexp-master/LocomotionAssayGeneration123/LocoAssayG123.csv&quot;, show_col_types = F)
female_loco &lt;- loco %&gt;% filter(sex == &quot;f&quot;)
male_loco &lt;- loco %&gt;% filter(sex == &quot;m&quot;)

# CFM &gt; FLX for activity (on-sig), which is the prediction (so we need a positive effect size)
female_loco_model &lt;- glm((cbind(active, inactive)) ~ regime + rep_population, 
                          data = female_loco, family = &quot;binomial&quot;)

# calculate effect of regime on % active flies (as a log odds ratio) separately for each sex
LOR_effects &lt;- contrast(emmeans(female_loco_model, ~ regime), method = &quot;pairwise&quot;)[1,] %&gt;% 
  as_tibble() %&gt;% 
  rename(logOR = estimate) %&gt;% 
  mutate(logOR = -1 * logOR) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Female locomotion&quot;,
       Generations = 123,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that females carrying female-evolved X chromosomes are less active than females carrying the control FM genotype (there is no difference, though).&quot;,
       Estimate = LOR_effects$Estimate,
       Var = LOR_effects$Var,
       Lower_95_CI = LOR_effects$Lower_95_CI,
       Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_5.csv&quot;)</code></pre>
</div>
<div id="male-locomotion-1" class="section level3">
<h3>Male locomotion</h3>
<pre class="r"><code># CFM &lt; FLX for activity (and it&#39;s significant), which is the opposite of the prediction (so we need a negative effect size)
male_loco_model &lt;- glm((cbind(active, inactive)) ~ regime + rep_population, 
                          data = male_loco, family = &quot;binomial&quot;)

# calculate effect of regime on % active flies (as a log odds ratio) separately for each sex
LOR_effects &lt;- contrast(emmeans(male_loco_model, ~ regime), method = &quot;pairwise&quot;)[1,] %&gt;% 
  as_tibble() %&gt;% 
  rename(logOR = estimate) %&gt;% 
  mutate(logOR = -1 * logOR) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)</code></pre>
<pre><code>Warning in (function (object, at, cov.reduce = mean, cov.keep = get_emm_option(&quot;cov.keep&quot;), : There are unevaluated constants in the response formula
Auto-detection of the response transformation may be incorrect</code></pre>
<pre class="r"><code>tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Male locomotion&quot;,
       Generations = 123,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that males carrying female-evolved X chromosomes are less active than males carrying the control FM genotype (actually it&#39;s significant in the opposite direction).&quot;,
       Estimate = LOR_effects$Estimate,
       Var = LOR_effects$Var,
       Lower_95_CI = LOR_effects$Lower_95_CI,
       Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_6.csv&quot;)</code></pre>
</div>
<div id="female-thorax-size" class="section level3">
<h3>Female thorax size</h3>
<pre class="r"><code>thorax &lt;- read_csv(&quot;input_data/lundhansen_github_data_FLXevoexp-master/ThoraxSizeGeneration72/ThoraxSizeG72.csv&quot;, show_col_types = F)
female_thorax &lt;- thorax %&gt;% 
  filter(sex == &quot;f&quot;) %&gt;% 
  mutate(size_mm = as.numeric(scale(size_mm)))
male_thorax &lt;- thorax %&gt;% 
  filter(sex == &quot;m&quot;) %&gt;% 
  mutate(size_mm = as.numeric(scale(size_mm)))

# FLX &gt; CFM, which is predicted since large size is a feminine trait 
model_female_thorax &lt;- lm(size_mm ~ regime + rep_population, data = female_thorax)

coef &lt;- coefficients(summary(model_female_thorax))

effects &lt;- tibble(Estimate = coef[2,1] * -1,
                  Var = coef[2,2] ^ 2,
                  Lower_95_CI = Estimate - coef[2,2]*1.96,
                  Upper_95_CI = Estimate + coef[2,2]*1.96)


tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Female thorax size&quot;,
       Generations = 72,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;The positive effect indicates that females carrying female-evolved X chromosomes had larger thoraxes than females with the control FM genotype.&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_7.csv&quot;)</code></pre>
</div>
<div id="male-thorax-size" class="section level3">
<h3>Male thorax size</h3>
<pre class="r"><code># FLX &gt; CFM, which is predicted since large size is a feminine trait 
model_male_thorax &lt;- lm(size_mm ~ regime + rep_population, data = male_thorax)

coef &lt;- coefficients(summary(model_male_thorax))

effects &lt;- tibble(Estimate = coef[2,1] * -1,
                  Var = coef[2,2] ^ 2,
                  Lower_95_CI = Estimate - coef[2,2]*1.96,
                  Upper_95_CI = Estimate + coef[2,2]*1.96)


tibble(Study = &quot;Lund-Hansen et al. (2020) Evolution 74, 2703–2713&quot;,
       Trait = &quot;Male thorax size&quot;,
       Generations = 72,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = &quot;A positive effect would mean that males carrying female-evolved X chromosomes had larger thoraxes than males with the control FM genotype (there is no difference, though).&quot;,
       Estimate = effects$Estimate,
       Var = effects$Var,
       Lower_95_CI = effects$Lower_95_CI,
       Upper_95_CI = effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/lundhansen_8.csv&quot;)</code></pre>
</div>
</div>
<div id="manat-et-al.-2021" class="section level2">
<h2>Manat et al. (2021)</h2>
<p>Manat, Y., Lund-Hansen, K. K., Katsianis, G., &amp; Abbott, J. K.
(2021). Female-limited X-chromosome evolution effects on male pre- and
post-copulatory success. <em>Biology Letters</em>, 17, 20200915.</p>
<div id="male-mating-success-1" class="section level3">
<h3>Male mating success</h3>
<div id="manat-et-al.s-original-data" class="section level4">
<h4>Manat et al.’s original data</h4>
<p>The data were extracted from Figure 2a using PlotDigitizer.</p>
<pre class="r"><code>df &lt;- tibble(
  treatment = c(&quot;FLX&quot;, &quot;FLX&quot;, &quot;CFM&quot;, &quot;CFM&quot;),
  status = c(&quot;unmated&quot;, &quot;mated&quot;, &quot;unmated&quot;, &quot;mated&quot;),
  n = c(4,76,10,70)) %&gt;% 
  pivot_wider(names_from = status, values_from = n)

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
unmated
</th>
<th style="text-align:right;">
mated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FLX
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
76
</td>
</tr>
<tr>
<td style="text-align:left;">
CFM
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
70
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-16" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>as.matrix(df[,2:3]) %&gt;% 
  get_effect_size_2x2_matrix() %&gt;% 
  mutate(Study = &quot;Manat et al. (2021) Biol. Lett. 17, 20200915&quot;, 
         Trait = &quot;Male mating success&quot;, 
         Generations = &quot;107&quot;,
         Organism = &quot;Drosophila melanogaster&quot;,
         Interpretation = &quot;A positive effect size would mean that males carrying an X chromosome that underwent female-limited evolution would have lower mating success compared to males carrying control X chromosomes (however, there&#39;s a significant difference in the opposite direction).&quot;,
         .before = 1) %&gt;% 
  write_csv(&quot;effect_sizes/manat_1.csv&quot;)</code></pre>
</div>
</div>
<div id="mating-speed-1" class="section level3">
<h3>Mating speed</h3>
<div id="manat-et-al.s-original-data-1" class="section level4">
<h4>Manat et al.’s original data</h4>
<p>The data were extracted from Figure 2b using PlotDigitizer.</p>
<pre class="r"><code>df &lt;- tibble(
  treatment = c(&quot;FLX&quot;, &quot;CFM&quot;),
  mean = c(10.14735444556975, 12.273534242844619),
  SE = c(11.07794959957106 - 10.14735444556975, 
         13.353237048825946 - 12.273534242844619),
  n = c(80,80))

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FLX
</td>
<td style="text-align:right;">
10.14735
</td>
<td style="text-align:right;">
0.9305952
</td>
<td style="text-align:right;">
80
</td>
</tr>
<tr>
<td style="text-align:left;">
CFM
</td>
<td style="text-align:right;">
12.27353
</td>
<td style="text-align:right;">
1.0797028
</td>
<td style="text-align:right;">
80
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-17" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>mes(m.1 = df$mean[1],
    m.2 = df$mean[2],
    sd.1 = df$SE[1] * sqrt(80),
    sd.2 = df$SE[2] * sqrt(80),
    n.1 = 880, n.2 = 80, verbose = F) %&gt;% 
  parse_es(study = &quot;Manat et al. (2021) Biol. Lett. 17, 20200915&quot;, 
           trait = &quot;Male mating speed&quot;, 
           generations = &quot;107&quot;,
           organism = &quot;Drosophila melanogaster&quot;,
           interpretation = &quot;A positive effect size would mean that males carrying an X chromosome that underwent female-limited evolution would take longer to induce females to mate compared to males carrying control X chromosomes (however, there is a significant difference in the opposite direction).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/manat_2.csv&quot;)</code></pre>
</div>
</div>
<div id="male-sperm-offence" class="section level3">
<h3>Male sperm offence</h3>
<div id="manat-et-al.s-original-data-2" class="section level4">
<h4>Manat et al.’s original data</h4>
<p>The data were extracted from Figure 2c using PlotDigitizer.</p>
<pre class="r"><code>df &lt;- tibble(
  treatment = c(&quot;FLX&quot;, &quot;CFM&quot;),
  mean = c(0.21776979298517823, 0.1056360840036476),
  SE = c(0.26085500796210853 - 0.21776979298517823, 
         0.1243885508962476 - 0.1056360840036476),
  n = c(123,123))

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FLX
</td>
<td style="text-align:right;">
0.2177698
</td>
<td style="text-align:right;">
0.0430852
</td>
<td style="text-align:right;">
123
</td>
</tr>
<tr>
<td style="text-align:left;">
CFM
</td>
<td style="text-align:right;">
0.1056361
</td>
<td style="text-align:right;">
0.0187525
</td>
<td style="text-align:right;">
123
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-18" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>mes(m.1 = df$mean[2],
    m.2 = df$mean[1],
    sd.1 = df$SE[2] * sqrt(80),
    sd.2 = df$SE[1] * sqrt(80),
    n.1 = 880, n.2 = 80, verbose = F) %&gt;% 
  parse_es(study = &quot;Manat et al. (2021) Biol. Lett. 17, 20200915&quot;, 
           trait = &quot;Male sperm offence&quot;, 
           generations = &quot;160&quot;,
           organism = &quot;Drosophila melanogaster&quot;,
           interpretation = &quot;A positive effect size would mean that males carrying an X chromosome that underwent female-limited evolution would have worse sperm offence (i.e. sire fewer offspring with previously mated females) compared to males carrying control X chromosomes (however the finding is significant in the opposite direction).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/manat_3.csv&quot;)</code></pre>
</div>
</div>
<div id="male-sperm-defence" class="section level3">
<h3>Male sperm defence</h3>
<div id="manat-et-al.s-original-data-3" class="section level4">
<h4>Manat et al.’s original data</h4>
<p>The data were extracted from Figure 2d using PlotDigitizer.</p>
<pre class="r"><code>df &lt;- tibble(
  treatment = c(&quot;FLX&quot;, &quot;CFM&quot;),
  mean = c(0.8606096836369432, 0.9267285480341434),
  SE = c(0.8815000797066794 - 0.8606096836369432, 
         0.9407380838514268 - 0.9267285480341434),
  n = c(123,123))

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FLX
</td>
<td style="text-align:right;">
0.8606097
</td>
<td style="text-align:right;">
0.0208904
</td>
<td style="text-align:right;">
123
</td>
</tr>
<tr>
<td style="text-align:left;">
CFM
</td>
<td style="text-align:right;">
0.9267285
</td>
<td style="text-align:right;">
0.0140095
</td>
<td style="text-align:right;">
123
</td>
</tr>
</tbody>
</table>
</div>
<div id="calculating-effect-size-19" class="section level4">
<h4>Calculating effect size</h4>
<pre class="r"><code>mes(m.1 = df$mean[2],
    m.2 = df$mean[1],
    sd.1 = df$SE[2] * sqrt(80),
    sd.2 = df$SE[1] * sqrt(80),
    n.1 = 880, n.2 = 80, verbose = F) %&gt;% 
  parse_es(study = &quot;Manat et al. (2021) Biol. Lett. 17, 20200915&quot;, 
           trait = &quot;Male sperm defence&quot;, 
           generations = &quot;160&quot;,
           organism = &quot;Drosophila melanogaster&quot;,
           interpretation = &quot;The positive effect indicates that males carrying an X chromosome that underwent female-limited evolution have worse sperm defence (i.e. sire fewer offspring when after the female remates) compared to males carrying control X chromosomes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/manat_4.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="grieshop-et-al.-2025" class="section level2">
<h2>Grieshop et al. (2025)</h2>
<p>Grieshop, K., Liu, M. J., Frost, R. S., Lindsay, M. P., Bayoumi, M.,
Brengdahl, M. I., Molnar, R. I., &amp; Agrawal, A. F. (2025). Expression
divergence in response to sex-biased selection. <em>Molecular Biology
and Evolution</em>, 42(7) msaf099.</p>
<p>Here, I obtained the raw data from the authors, and re-ran a binomial
GLMM that followed the original analysis as closely as I could figure
out. The model measures the probability of mating (vs not mating) for
males and females and flies carrying ‘Red’ chromosomes (copies of
Chromosome 2 which spent most of their recent evolutionary history in
males) versus ‘Non-Red’ chromosomes (which spent a majority of
generations in females). From the binomial GLM I get the treatment
effect size as a log odds ratio, and then convert that to Cohen’s d.</p>
<pre class="r"><code>###### This block of code copied from the authors&#39; archived R script ########################################################################

#Load in fitness data
DsRed_Data &lt;- read.csv(&quot;input_data/grieshop2025_DsRed_Data_Formatted.csv&quot;)

# Data Prep
#######
# Set up a vector full of zeroes of the appropriate length, then...
Prop_Mated &lt;- rep.int(0, dim(DsRed_Data)[1])

# iterate over each observation in the DsRed_Data object
for (i in 1:dim(DsRed_Data)[1]){
  # if the observation is Male for that population replicate cage and generation
  if (DsRed_Data$Sex[i] == &quot;Male&quot;){
    # the proportion of mated males 
    Prop_Mated[i] &lt;- DsRed_Data$Male_Prop_Mated[i]
  }
  else{
    # the proportion of mated females for that population replicate cage and generation
    Prop_Mated[i] &lt;- DsRed_Data$Female_Prop_Mated[i]
  }
  #... fill in said vector at any given point with the appropriate value 
  # for the proportion of sampled individuals of the given sex found to be mated...
}

#...and then add that vector as another column.
DsRed_Data &lt;- cbind(DsRed_Data, Prop_Mated) 

markerXpop.bothsex.model &lt;- glmer(
  Mating_Status == &quot;Mated&quot; ~ Marker*Sex + (Marker|Population),
  binomial(link=&#39;probit&#39;), offset = Prop_Mated, data = DsRed_Data)

###### End of copied code ####################################

# calculate effect of marker on mating rate (as a log odds ratio) separately for each sex
LOR_effects &lt;- contrast(emmeans(markerXpop.bothsex.model, ~ Marker | Sex), method = &quot;pairwise&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(estimate = c(1,-1) * estimate) %&gt;% # change sign of the effect size for males
  rename(logOR = estimate) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi, # convert LOR to Cohen&#39;s d
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

tibble(Study = &quot;Grieshop et al. (2025) Molec. Biol. Evol. 42 msaf099.&quot;,
       Trait = c(&quot;Female mating rate&quot;, &quot;Male mating rate&quot;),
       Generations = 103,
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = c(
         &quot;A positive effect would mean that females carrying primarily male-evolved copies of chromosome 2 mate less often, compared to females with primarily female-evolved copies of chromosome 2 (there is no difference, though).&quot;, 
         &quot;This positive effect means that males carrying primarily male-evolved copies of chromosome 2 mate more often, compared to males with primarily female-evolved copies of chromosome 2.&quot;),
       Estimate = LOR_effects$Estimate,
       Var = LOR_effects$Var,
       Lower_95_CI = LOR_effects$Lower_95_CI,
       Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/grieshop2025.csv&quot;)</code></pre>
</div>
<div id="melo-gavin-et-al.-2025" class="section level2">
<h2>Melo-Gavin et al. (2025)</h2>
<p>Melo-Gavin, C., Lindsay, M. P., &amp; Agrawal, A. F. (2025). Response
of mate harm to sex-separated gene pools. bioRxiv, 2025-09.</p>
<div id="melo-gavin-et-al.s-raw-data" class="section level3">
<h3>Melo-Gavin et al.’s raw data</h3>
<p>The data were obtained from the left two bars in Figure 1, using
PlotDigitizer. The figure shows means and 95% CIs, so I calculated the
standard deviation from the CIs as
<code>(CI_95 / 1.96) * sqrt(N)</code>, i.e. assuming the CIs were
calculated as 1.96 times the standard error of the mean.</p>
<pre class="r"><code>df &lt;- tibble(male_type = c(&quot;NonRed&quot;, &quot;Red&quot;),
       line_type = c(&quot;Treatment&quot;, &quot;Treatment&quot;),
       mean = c(57.18512618727785, 53.99935806754556),
       CI_95 = c(59.0035781790516 - 57.18512618727785,
                 55.64306178005492 - 53.99935806754556),
       N = c(180, 180),
       SE = (CI_95 / 1.96),
       SD = SE * sqrt(N))

mes(m.1 = df$mean[df$male_type==&quot;NonRed&quot;],
    m.2 = df$mean[df$male_type==&quot;Red&quot;],
    sd.1 = df$SD[df$male_type==&quot;NonRed&quot;],
    sd.2 = df$SD[df$male_type==&quot;Red&quot;],
    n.1 = 180, n.2 = 180, verbose=F) %&gt;% 
  parse_es(study = &quot;Melo-Gavin et al. (2025) bioRxiv, https://doi.org/10.1101/2025.09.29.678620.&quot;,
           generations = 11,
           organism = &quot;Drosophila melanogaster&quot;,
           trait = &quot;Male harm to females&quot;,
           interpretation = &quot;The positive effect indicates that males carrying male-evolved chromosomes reduced the fecundity of their female mates more than did males carrying chromosomes that had undergone female-biased evolution.&quot;
          ) %&gt;% 
  write_csv(&quot;effect_sizes/melo_gavin2025_maleharm.csv&quot;)</code></pre>
</div>
</div>
<div id="thyagarajan-et-al.-2025" class="section level2">
<h2>Thyagarajan et al. (2025)</h2>
<p>Thyagarajan, H., Sayyed, I., Baroody, M. G., Kowal, J. A., Day, T.,
&amp; Chippindale, A. K. (2025). Mixed evidence for intralocus sexual
conflict from male-limited selection in <em>Drosophila
melanogaster</em>. <em>Journal of Heredity</em>, esaf072.</p>
<div id="female-and-male-fitness" class="section level3">
<h3>Female and male fitness</h3>
<p>These variables are referred to as ‘competitive reproductive fitness’
in the paper. I obtained the raw data from Dryad and ran one model for
each sex. The models were binomial GLMs with treatment (male-evolved or
control), replicate, and generation as fixed effects (no interactions
are fitted, i.e. I estimate the average treatment effect across the 3
starting genomes, labelled ‘replicates’). From the binomial GLM I get
the treatment effect size as a log odds ratio, and then convert that to
Cohen’s d.</p>
<pre class="r"><code>female_CRF &lt;- read_csv(&quot;input_data/Thyagarajan_2025_dryad_data/F_CRF_wt.cr.csv&quot;,
                       show_col_types = F) %&gt;% 
  mutate(Sex = &quot;F&quot;, Replicate = factor(Replicate))
male_CRF &lt;- read_csv(&quot;input_data/Thyagarajan_2025_dryad_data/M_CRF_wt.cr.csv&quot;,
                     show_col_types = F) %&gt;% 
  mutate(Sex = &quot;M&quot;, Replicate = factor(Replicate)) %&gt;% 
  filter(Treatment %in% c(&quot;MLSD&quot;, &quot;MC&quot;))


# calculate effect of marker on mating rate (as a log odds ratio) separately for each sex
## GLM for females:
model &lt;- glm(cbind(Red, Brown) ~ Treatment + Replicate + Gen, data = female_CRF, family = &quot;quasibinomial&quot;)
female_effects &lt;- contrast(emmeans(model, ~ Treatment), method = &quot;pairwise&quot;) %&gt;% 
  as_tibble() %&gt;% 
  rename(logOR = estimate) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

## GLM for males:
model2 &lt;- glm(cbind(Red, Brown) ~ Treatment + Replicate + Gen, data = male_CRF, family = &quot;quasibinomial&quot;)
male_effects &lt;- contrast(emmeans(model2, ~ Treatment), method = &quot;pairwise&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(estimate = estimate * -1) %&gt;% # change sign of effect size (it goes in the predicted direction)
  rename(logOR = estimate) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

LOR_effects &lt;- bind_rows(female_effects, male_effects)

tibble(Study = &quot;Thyagarajan et al. (2025) J. Hered. esaf072&quot;,
       Trait = c(&quot;Female fitness&quot;, &quot;Male fitness&quot;),
       Generations = c(50, 64),
       Organism = &quot;Drosophila melanogaster&quot;,
       Interpretation = c(&quot;This positive effect means that females carrying male-evolved haplotypes have lower fitness than females carrying control genomes.&quot;, 
                          &quot;This positive effect means that males carrying male-evolved haplotypes have higher fitness than males carrying control genomes.&quot;),
       Estimate = LOR_effects$Estimate,
       Var = LOR_effects$Var,
       Lower_95_CI = LOR_effects$Lower_95_CI,
       Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/thyagarajan2025_1.csv&quot;)</code></pre>
</div>
<div id="male-mating-success-2" class="section level3">
<h3>Male mating success</h3>
<pre class="r"><code>mating_competition &lt;- read_csv(&quot;input_data/Thyagarajan_2025_dryad_data/M_MC_wt.cr.csv&quot;,
                               show_col_types = F) %&gt;% 
  filter(!is.na(Mated.male)) %&gt;% 
  group_by(Selection, Replicate) %&gt;% 
  summarise(n_wt = sum(Mated.male == &quot;wt&quot;),
            n_bw = sum(Mated.male == &quot;bw&quot;))

model &lt;- glm(cbind(n_wt, n_bw) ~ Selection + Replicate, data = mating_competition, family = &quot;binomial&quot;)

LOR_effects &lt;- contrast(emmeans(model, ~ Selection), method = &quot;pairwise&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(estimate = estimate * -1) %&gt;% # change sign of effect size (it goes in the predicted direction)
  rename(logOR = estimate) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

tibble(
  Study = &quot;Thyagarajan et al. (2025) J. Hered. esaf072&quot;,
  Trait = &quot;Male mating success&quot;,
  Generations = 64,
  Organism = &quot;Drosophila melanogaster&quot;,
  Interpretation = &quot;A positive effectwould mean that males carrying male-evolved haplotypes are more likely to mate with females (over standard male competitors) compared to males carrying control genomes (there is no signficant difference).&quot;,
  Estimate = LOR_effects$Estimate,
  Var = LOR_effects$Var,
  Lower_95_CI = LOR_effects$Lower_95_CI,
  Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/thyagarajan2025_2.csv&quot;)</code></pre>
</div>
<div id="male-sperm-offence-1" class="section level3">
<h3>Male sperm offence</h3>
<pre class="r"><code>sperm_comp_data &lt;- read_csv(&quot;input_data/Thyagarajan_2025_dryad_data/M_SC_wt.cr.csv&quot;,
                            show_col_types = F) %&gt;% 
  filter(Treatment %in% c(&quot;MC&quot;, &quot;MLSD&quot;)) %&gt;% 
  mutate(Replicate = factor(Replicate))

model &lt;- glm(cbind(WT, U) ~ Treatment + Replicate, family = &quot;binomial&quot;, data = sperm_comp_data)
LOR_effects &lt;- contrast(emmeans(model, ~ Treatment), method = &quot;pairwise&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(estimate = estimate * -1) %&gt;% # change sign of effect size (it goes in the predicted direction)
  rename(logOR = estimate) %&gt;% 
  mutate(Estimate = logOR * sqrt(3) / pi,
         Var = (SE * sqrt(3) / pi)^2,
         Lower_95_CI = (logOR - SE * 1.96) * sqrt(3) / pi,
         Upper_95_CI = (logOR + SE * 1.96) * sqrt(3) / pi)

tibble(
  Study = &quot;Thyagarajan et al. (2025) J. Hered. esaf072&quot;,
  Trait = &quot;Male sperm offence&quot;,
  Generations = 64,
  Organism = &quot;Drosophila melanogaster&quot;,
  Interpretation = &quot;A positive effect would indicate that males carrying male-evolved haplotypes sire more offspring when mating with previously-mated females, compared to males carrying control genomes (there is no significant effect).&quot;,
  Estimate = LOR_effects$Estimate,
  Var = LOR_effects$Var,
  Lower_95_CI = LOR_effects$Lower_95_CI,
  Upper_95_CI = LOR_effects$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/thyagarajan2025_3.csv&quot;)</code></pre>
</div>
</div>
<div id="nordén-et-al.-2023" class="section level2">
<h2>Nordén et al. (2023)</h2>
<p>Nordén, A. K., Ramm, S. A., &amp; Abbott, J. K. (2023). Rapid
evolution of sex role specialization in a hermaphrodite under
sex-limited selection. <em>Evolution</em>, 77(4), 1066-1076.</p>
<p>The raw data were taken from the Dryad repository, and analysed using
the same model as in the original study (copying the R code). The
fitness data were mean-centered and scaled prior to analysis, such that
model-estimated differences in treatment means are equal to Cohen’s
d. </p>
<p>Note that this study has both a female-limited and a male-limited
evolution treatment, unlike all the others in this section, and it
measured both male and female fitness. To address this, I include 4
contrasts in the meta-analysis: female-limited evolution vs the control
for both sexes’ fitnesses, and male-limited evolution vs the control for
both sexes’ fitnesses. I predict elevated fitness in the selected sex
and reduced fitness in the non-selected sex, and adjust the effect size
signs to reflect this.</p>
<pre class="r"><code>fitness.dataCS &lt;- read.csv(&quot;input_data/norden_dryad_data/Mlig_fitness_data_C_standard.csv&quot;, 
                           sep=&quot;;&quot;, dec=&quot;,&quot;, header=TRUE, stringsAsFactors = T) %&gt;% 
  as_tibble() %&gt;% 
  mutate(Stand_Fitness = as.numeric(scale(Stand_Fitness)))

modfitCS &lt;- lmer(Stand_Fitness~Treatment*sex_role + (1|ID) + (1|Treatment:sex_role:Population), data = fitness.dataCS)

cohens_d &lt;- emmeans(modfitCS, ~ Treatment | sex_role) %&gt;% 
  pairs() %&gt;% as_tibble() %&gt;% 
  filter(str_detect(contrast, &quot;Control&quot;)) %&gt;% 
  select(trait = contrast, Estimate = estimate, SE)

cohens_d$trait &lt;- c(
  &quot;Female fitness (female-limited lines)&quot;,
  &quot;Female fitness (male-limited lines)&quot;,
  &quot;Male fitness (female-limited lines)&quot;,
  &quot;Male fitness (male-limited lines)&quot;)

cohens_d$Estimate &lt;- cohens_d$Estimate * c(-1, 1, 1, -1) 

cohens_d$Lower_95_CI &lt;- cohens_d$Estimate - cohens_d$SE * 1.96
cohens_d$Upper_95_CI &lt;- cohens_d$Estimate + cohens_d$SE * 1.96

tibble(Study = &quot;Nordén et al. (2020) Evolution 77, 1066-1076&quot;,
       Trait = cohens_d$trait,
       Generations = 14,
       Organism = &quot;Macrostomum lignano&quot;,
       Interpretation = c(
         &quot;The positive effect size means that worms carrying the feminising allele had higher fitness when breeding in the female sex role compared to worms carrying the control allele.&quot;,
         &quot;A positive effect would mean that worms carrying the masculinising allele had lower fitness when breeding in the female sex role compared to worms carrying the control allele (actually, they had non-significantly higher fitness).&quot;,
         &quot;A positive effect would mean that worms carrying the feminising allele had lower fitness when breeding in the male sex role compared to worms carrying the control allele (they did, but the difference is not significant).&quot;,
         &quot;A positive effect would mean that worms carrying the masculinising allele had higher fitness when breeding in the male sex role compared to worms carrying the control allele (they did, but the difference is not significant).&quot;
       ),
       Estimate = cohens_d$Estimate,
       Var = cohens_d$SE ^ 2,
       Lower_95_CI = cohens_d$Lower_95_CI,
       Upper_95_CI = cohens_d$Upper_95_CI) %&gt;% 
  write_csv(&quot;effect_sizes/norden2020.csv&quot;)</code></pre>
<!-- ## Melo-Gavin et al. (2026) -->
<!-- Melo-Gavin, C., Liu, M. J., & Agrawal, A. F. (2026). Local Adaptation to the Sex Environment: Reciprocal Sex-Limited Selection in Different Thermal Regimes. bioRxiv, 2026-01. https://doi.org/10.64898/2026.01.23.701411 -->
<!-- ### Female fitness -->
<!-- #### Melo-Gavin et al.'s original data -->
<!-- ```{r} -->
<!-- df <- tibble( -->
<!--   chr_type = c("female-limited", "male-limited"), -->
<!--   mean = c(1.0764962187609557, 0.9991861571593129), -->
<!--   SE = c(1.0956453147693694 - 1.0764962187609557,  -->
<!--          1.0255621775930286 - 0.9991861571593129), -->
<!--   n = c(??,??)) -->
<!-- kable(df) %>%  -->
<!--   kable_styling(full_width = F) -->
<!-- Melo-Gavin et al. (2026). bioRxiv, https://doi.org/10.64898/2026.01.23.701411 -->
<!-- ``` -->
</div>
</div>
<div id="middle-class-neighbourhood-experiments" class="section level1">
<h1>Middle-class neighbourhood experiments</h1>
<p>This section covers publications that used a sex-specific middle
class neighbourhood to relax selection on one sex. In general, I predict
reduced fitness for the unselected sex, increased fitness in the other
sex (due to the spread of sexually antagonistic alleles benefitting the
selected sex), and phenotypic evolution towards the typical trait values
for the selected sex (e.g. in <em>Drosophila</em>, chromosomes that had
undergone male-limited evolution in should encode smaller body size and
slower development time). The effect size signs selected reflect these
expectations.</p>
<div id="radwan-et-al.-2004" class="section level2">
<h2>Radwan et al. (2004)</h2>
<p>Radwan, J., Unrug, J., Snigórska, K., &amp; Gawrońska, K. (2004).
Effectiveness of sexual selection in preventing fitness deterioration in
bulb mite populations under relaxed natural selection. <em>Journal of
Evolutionary Biology</em>, 17(1), 94–99.</p>
<div id="radwans-original-data" class="section level3">
<h3>Radwan’s original data</h3>
<p>The means and standard errors for each variable and treatment group
were extracted from Table 1. Sample sizes for individual treatments are
not given, so I divided the total sample size by 3 and rounded off to
estimate them.</p>
<pre class="r"><code>df &lt;- tibble(
  Variable = rep(c(&quot;Viability&quot;,
                   &quot;Female longevity&quot;,
                   &quot;Female fecundity&quot;,
                   &quot;Male fitness&quot;),2),
  
  Treatment = c(rep(&quot;SS&quot;, 4), rep(&quot;C&quot;, 4)),
  
  mean = c(0.761, 23.74, 108.7, 0.543, 0.733, 25.35, 131.7, 0.596),
  SE   = c(0.102, 2.45, 4.47, 0.077, 0.066, 1.17, 9.51, 0.061),
  n = round(rep(c(161/3, 136/3, 139/3, 100/3), 2))
)

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Variable
</th>
<th style="text-align:left;">
Treatment
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Viability
</td>
<td style="text-align:left;">
SS
</td>
<td style="text-align:right;">
0.761
</td>
<td style="text-align:right;">
0.102
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
Female longevity
</td>
<td style="text-align:left;">
SS
</td>
<td style="text-align:right;">
23.740
</td>
<td style="text-align:right;">
2.450
</td>
<td style="text-align:right;">
45
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fecundity
</td>
<td style="text-align:left;">
SS
</td>
<td style="text-align:right;">
108.700
</td>
<td style="text-align:right;">
4.470
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
SS
</td>
<td style="text-align:right;">
0.543
</td>
<td style="text-align:right;">
0.077
</td>
<td style="text-align:right;">
33
</td>
</tr>
<tr>
<td style="text-align:left;">
Viability
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.733
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
Female longevity
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
25.350
</td>
<td style="text-align:right;">
1.170
</td>
<td style="text-align:right;">
45
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fecundity
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
131.700
</td>
<td style="text-align:right;">
9.510
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.596
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
33
</td>
</tr>
</tbody>
</table>
</div>
<div id="viability" class="section level3">
<h3>Viability</h3>
<pre class="r"><code>mes(m.2 = df$mean[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Viability&quot;],
    m.1 = df$mean[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Viability&quot;],
    sd.2 = df$SE[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Viability&quot;] * 
      sqrt(df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Viability&quot;]),
    sd.1 = df$SE[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Viability&quot;] * 
      sqrt(df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Viability&quot;]),
    n.2 = df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Viability&quot;],
    n.1 = df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Viability&quot;], verbose=F) %&gt;% 
  parse_es(study = &quot;Radwan et al. (2004) J. Evol. Biol. 17, 94–99.&quot;,
           generations = 11,
           organism = &quot;Rhizoglyphus robini&quot;,
           trait = &quot;Viability&quot;,
           interpretation = &quot;A positive effect size would indicate lower viability in populations that experienced relaxed selection on females relative to those experiencing selection on both sexes (the difference is non-significant and in the opposite direction).&quot;
          ) %&gt;% 
  write_csv(&quot;effect_sizes/radwan2004_1.csv&quot;)</code></pre>
</div>
<div id="female-longevity" class="section level3">
<h3>Female longevity</h3>
<pre class="r"><code>mes(m.2 = df$mean[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female longevity&quot;],
    m.1 = df$mean[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female longevity&quot;],
    sd.2 = df$SE[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female longevity&quot;] * 
      sqrt(df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female longevity&quot;]),
    sd.1 = df$SE[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female longevity&quot;] * 
      sqrt(df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female longevity&quot;]),
    n.2 = df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female longevity&quot;],
    n.1 = df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female longevity&quot;], verbose=F) %&gt;% 
  parse_es(study = &quot;Radwan et al. (2004) J. Evol. Biol. 17, 94–99.&quot;,
           generations = 11,
           organism = &quot;Rhizoglyphus robini&quot;,
           trait = &quot;Female longevity&quot;,
           interpretation = &quot;A positive effect size would indicate reduced female longevity in populations that experienced relaxed selection on females relative to those experiencing selection on both sexes (the difference is non-significant and in the predicted direction).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/radwan2004_2.csv&quot;)</code></pre>
</div>
<div id="female-fecundity" class="section level3">
<h3>Female fecundity</h3>
<pre class="r"><code>mes(m.2 = df$mean[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female fecundity&quot;],
    m.1 = df$mean[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female fecundity&quot;],
    sd.2 = df$SE[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female fecundity&quot;] * 
      sqrt(df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female fecundity&quot;]),
    sd.1 = df$SE[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female fecundity&quot;] * 
      sqrt(df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female fecundity&quot;]),
    n.2 = df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Female fecundity&quot;],
    n.1 = df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Female fecundity&quot;], verbose=F) %&gt;% 
  parse_es(study = &quot;Radwan et al. (2004) J. Evol. Biol. 17, 94–99.&quot;,
           generations = 11,
           organism = &quot;Rhizoglyphus robini&quot;,
           trait = &quot;Female fecundity&quot;,
           interpretation = &quot;The positive effect size indicates reduced fecundity in populations that experienced relaxed selection on females relative to those experiencing selection on both sexes.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/radwan2004_3.csv&quot;)</code></pre>
</div>
<div id="male-fitness-3" class="section level3">
<h3>Male fitness</h3>
<pre class="r"><code>mes(m.1 = df$mean[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Male fitness&quot;],
    m.2 = df$mean[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Male fitness&quot;],
    sd.1 = df$SE[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Male fitness&quot;] * 
      sqrt(df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Male fitness&quot;]),
    sd.2 = df$SE[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Male fitness&quot;] * 
      sqrt(df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Male fitness&quot;]),
    n.1 = df$n[df$Treatment==&quot;SS&quot; &amp; df$Variable == &quot;Male fitness&quot;],
    n.2 = df$n[df$Treatment==&quot;C&quot; &amp; df$Variable == &quot;Male fitness&quot;], verbose=F) %&gt;% 
  parse_es(study = &quot;Radwan et al. (2004) J. Evol. Biol. 17, 94–99.&quot;,
           generations = 11,
           organism = &quot;Rhizoglyphus robini&quot;,
           trait = &quot;Male fitness&quot;,
           interpretation = &quot;A positive effect size would indicate increased fitness in populations that experienced relaxed selection on females relative to those experiencing selection on both sexes (the difference is non-significant and in the opposite direction).&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/radwan2004_4.csv&quot;)</code></pre>
</div>
</div>
<div id="morrow-et-al.-2008" class="section level2">
<h2>Morrow et al. (2008)</h2>
<p>Morrow, E. H., Stewart, A. D., &amp; Rice, W. R. (2008). Assessing
the extent of genome-wide intralocus sexual conflict via experimentally
enforced gender-limited selection. <em>Journal of Evolutionary
Biology</em>, 21(4), 1046–1054.</p>
<div id="morrow-et-al.s-original-data" class="section level3">
<h3>Morrow et al.’s original data</h3>
<pre class="r"><code>df &lt;- tibble(treatment = c(&quot;selected sex&quot;, &quot;non-selected sex&quot;,&quot;selected sex&quot;, &quot;non-selected sex&quot;),
       trait = c(&quot;male fitness&quot;, &quot;male fitness&quot;, &quot;female fitness&quot;, &quot;female fitness&quot;),
       mean = c(9.761182759612618, 8.5105834691725, 38.60211669355156, 33.71168902702915),
       SE = c(10.07670917633522 - 9.761182759612618,
              8.812266276728353 - 8.5105834691725,
              38.99686561487718 - 38.60211669355156,
              33.985783792999435 - 33.71168902702915),
       N = c(60, 60, 60, 60))

kable(df) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
trait
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
N
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
selected sex
</td>
<td style="text-align:left;">
male fitness
</td>
<td style="text-align:right;">
9.761183
</td>
<td style="text-align:right;">
0.3155264
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
non-selected sex
</td>
<td style="text-align:left;">
male fitness
</td>
<td style="text-align:right;">
8.510583
</td>
<td style="text-align:right;">
0.3016828
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
selected sex
</td>
<td style="text-align:left;">
female fitness
</td>
<td style="text-align:right;">
38.602117
</td>
<td style="text-align:right;">
0.3947489
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
non-selected sex
</td>
<td style="text-align:left;">
female fitness
</td>
<td style="text-align:right;">
33.711689
</td>
<td style="text-align:right;">
0.2740948
</td>
<td style="text-align:right;">
60
</td>
</tr>
</tbody>
</table>
</div>
<div id="male-fitness-effect-size-2" class="section level3">
<h3>Male fitness effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[df$trait == &quot;male fitness&quot; &amp; df$treatment == &quot;selected sex&quot;],
    m.2 = df$mean[df$trait == &quot;male fitness&quot; &amp; df$treatment == &quot;non-selected sex&quot;],
    sd.1 = df$SE[df$trait == &quot;male fitness&quot; &amp; df$treatment == &quot;selected sex&quot;] * sqrt(60),
    sd.2 = df$SE[df$trait == &quot;male fitness&quot; &amp; df$treatment == &quot;non-selected sex&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose = F) %&gt;% 
  parse_es(study = &quot;Morrow et al. (2008) J. Evol. Biol. 21, 1046–1054&quot;,
           trait = &quot;Male fitness&quot;, 
           generations = 26, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect sie indicates that male fitness was higher in lines where selection was relaxed on females compared to lines where selection was relaxed on males.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/morrow2008_1.csv&quot;)</code></pre>
</div>
<div id="female-fitness-effect-size-2" class="section level3">
<h3>Female fitness effect size</h3>
<pre class="r"><code>mes(m.1 = df$mean[df$trait == &quot;female fitness&quot; &amp; df$treatment == &quot;selected sex&quot;],
    m.2 = df$mean[df$trait == &quot;female fitness&quot; &amp; df$treatment == &quot;non-selected sex&quot;],
    sd.1 = df$SE[df$trait == &quot;female fitness&quot; &amp; df$treatment == &quot;selected sex&quot;] * sqrt(60),
    sd.2 = df$SE[df$trait == &quot;female fitness&quot; &amp; df$treatment == &quot;non-selected sex&quot;] * sqrt(60),
    n.1 = 60, n.2 = 60, verbose = F) %&gt;% 
  parse_es(study = &quot;Morrow et al. (2008) J. Evol. Biol. 21, 1046–1054&quot;,
           trait = &quot;Female fitness&quot;, 
           generations = 26, 
           organism = &quot;Drosophila melanogaster&quot;, 
           interpretation = &quot;The positive effect sie indicates that female fitness was higher in lines where selection was relaxed on males compared to lines where selection was relaxed on females.&quot;) %&gt;% 
  write_csv(&quot;effect_sizes/morrow2008_2.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.2 (2025-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Tahoe 26.2

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] compute.es_0.2-6 kableExtra_1.4.0 survival_3.8-3   emmeans_2.0.1   
 [5] lmerTest_3.2-0   car_3.1-5        carData_3.0-6    lme4_1.1-38     
 [9] Matrix_1.7-4     readxl_1.4.5     lubridate_1.9.4  forcats_1.0.1   
[13] stringr_1.6.0    dplyr_1.2.0      purrr_1.2.1      readr_2.1.6     
[17] tidyr_1.3.2      tibble_3.3.1     ggplot2_4.0.2    tidyverse_2.0.0 
[21] workflowr_1.7.2 

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1    viridisLite_0.4.2   farver_2.1.2       
 [4] S7_0.2.1            fastmap_1.2.0       promises_1.5.0     
 [7] digest_0.6.39       timechange_0.4.0    estimability_1.5.1 
[10] lifecycle_1.0.5     processx_3.8.6      magrittr_2.0.4     
[13] compiler_4.5.2      rlang_1.1.7         sass_0.4.10        
[16] tools_4.5.2         yaml_2.3.12         knitr_1.51         
[19] bit_4.6.0           xml2_1.5.2          RColorBrewer_1.1-3 
[22] abind_1.4-8         withr_3.0.2         numDeriv_2016.8-1.1
[25] grid_4.5.2          git2r_0.36.2        scales_1.4.0       
[28] MASS_7.3-65         cli_3.6.5           mvtnorm_1.3-3      
[31] crayon_1.5.3        rmarkdown_2.30      reformulas_0.4.4   
[34] generics_0.1.4      otel_0.2.0          rstudioapi_0.18.0  
[37] httr_1.4.7          tzdb_0.5.0          minqa_1.2.8        
[40] cachem_1.1.0        splines_4.5.2       parallel_4.5.2     
[43] cellranger_1.1.0    vctrs_0.7.1         boot_1.3-32        
[46] jsonlite_2.0.0      callr_3.7.6         hms_1.1.4          
[49] pbkrtest_0.5.5      bit64_4.6.0-1       Formula_1.2-5      
[52] systemfonts_1.3.1   jquerylib_0.1.4     glue_1.8.0         
[55] nloptr_2.2.1        ps_1.9.1            stringi_1.8.7      
[58] gtable_0.3.6        later_1.4.5         pillar_1.11.1      
[61] htmltools_0.5.9     R6_2.6.1            textshaping_1.0.4  
[64] Rdpack_2.6.5        rprojroot_2.1.1     vroom_1.7.0        
[67] evaluate_1.0.5      lattice_0.22-7      backports_1.5.0    
[70] rbibutils_2.4.1     broom_1.0.12        httpuv_1.6.16      
[73] bslib_0.10.0        Rcpp_1.1.1          svglite_2.2.2      
[76] coda_0.19-4.1       nlme_3.1-168        whisker_0.4.1      
[79] xfun_0.56           fs_1.6.6            getPass_0.2-4      
[82] pkgconfig_2.0.3    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
